<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Tina Lasisi" />


<title>Analysis v2</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Goza_AMY_CNV</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="analysis.html">V1 Analyses</a>
</li>
<li>
  <a href="analysisv2.html">V2 Analyses</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Analysis v2</h1>
<h4 class="author">Tina Lasisi</h4>
<h4 class="date">25 July, 2023</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-07-25
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Goza_AMY_CNV/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230616code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230616)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230616code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230616)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtinalasisiGozaAMYCNVtreeb5db62c16564a512ad4937891c3ee62e7802e59etargetblankb5db62ca">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/tree/b5db62c16564a512ad4937891c3ee62e7802e59e" target="_blank">b5db62c</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtinalasisiGozaAMYCNVtreeb5db62c16564a512ad4937891c3ee62e7802e59etargetblankb5db62ca"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/tree/b5db62c16564a512ad4937891c3ee62e7802e59e" target="_blank">b5db62c</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/AMY_CNV_data/

Unstaged changes:
    Modified:   analysis/analysisv2.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/analysisv2.Rmd</code>) and HTML
(<code>docs/analysisv2.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/b5db62c16564a512ad4937891c3ee62e7802e59e/analysis/analysisv2.Rmd" target="_blank">b5db62c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
<td>
Update analysis v2
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/Goza_AMY_CNV/b5db62c16564a512ad4937891c3ee62e7802e59e/docs/analysisv2.html" target="_blank">b5db62c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
<td>
Update analysis v2
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/Goza_AMY_CNV/eb2e32f709faaf9e78b11c8e22c1e29ade2d49bc/docs/analysisv2.html" target="_blank">eb2e32f</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
<td>
Update analysis v2
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/6e019773d11bd25f82f5d91be75b101a2da374d8/analysis/analysisv2.Rmd" target="_blank">6e01977</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
<td>
Update analyses
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/Goza_AMY_CNV/6e019773d11bd25f82f5d91be75b101a2da374d8/docs/analysisv2.html" target="_blank">6e01977</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
<td>
Update analyses
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/170c8ef65fdd1fd4c40944e49aab0563198a7516/analysis/analysisv2.Rmd" target="_blank">170c8ef</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-14
</td>
<td>
Added flavor perception analyses
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/Goza_AMY_CNV/170c8ef65fdd1fd4c40944e49aab0563198a7516/docs/analysisv2.html" target="_blank">170c8ef</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-14
</td>
<td>
Added flavor perception analyses
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/07ec9306556753cb5847b51e0594ab20e52e480d/analysis/analysisv2.Rmd" target="_blank">07ec930</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-12
</td>
<td>
Publish repo
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/Goza_AMY_CNV/07ec9306556753cb5847b51e0594ab20e52e480d/docs/analysisv2.html" target="_blank">07ec930</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-12
</td>
<td>
Publish repo
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="clusters" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Clusters</h1>
<pre class="r"><code>df_saliva &lt;- read_csv(&quot;data/df_saliva.csv&quot;) %&gt;% 
  rename(TP_Before_1 = Total_Protein_Before_1_ug_mL,
         TP_Before_2 = Total_Protein_Before_2_ug_mL,
         TP_After_1 = Total_Protein_After_1_ug_mL,
         TP_After_2 = Total_Protein_After_2_ug_mL,
         sAA_Before = sAA_Mean_Collection_Before_U_mL,
         sAA_After = sAA_Mean_Collection_After_U_mL,
         TP_Before = Total_Protein_Before_Mean_ug_mL,
         TP_After = Total_Protein_Mean_After_ug_mL)</code></pre>
<p>First, we will do a PCA using genetic and protein variables to see if
there are any clusters in our dataset.</p>
<pre class="r"><code>vars_include &lt;- c(&quot;Primer1_Diploid&quot;, &quot;Primer2_Diploid&quot;, &quot;TP_Before_1&quot;, &quot;TP_Before_2&quot;, &quot;TP_After_1&quot;, &quot;TP_After_2&quot;, &quot;TP_Change&quot;, &quot;sAA_Before&quot;, &quot;sAA_After&quot;, &quot;sAA_Change&quot;, &quot;Weight_Saliva_g_1&quot;, &quot;Weight_Saliva_g_2&quot;, &quot;Flow_Rate_1_g_min&quot;, &quot;Flow_Rate_2_g_min&quot;, &quot;Sample_ID&quot;)</code></pre>
<div id="na-values-omitted" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> NA values
omitted</h2>
<p>We will do the analysis first by omitting na values.</p>
<pre class="r"><code># Scale the Data
df_saliva_scaled &lt;- df_saliva %&gt;%
  select(all_of(vars_include)) %&gt;% 
  select(-Sample_ID) %&gt;% 
  na.omit() %&gt;% 
  mutate(across(everything(), ~ as.numeric(.) %&gt;% scale))

# Data unscaled
df_saliva_unscaled &lt;- df_saliva %&gt;%
  select(all_of(vars_include)) %&gt;% 
  select(-&quot;Sample_ID&quot;) %&gt;% 
  na.omit()

# Perform PCA
pca &lt;- df_saliva_scaled %&gt;%
  prcomp(center = FALSE, scale. = FALSE)

# Print the summary of the PCA object
summary(pca)</code></pre>
<pre><code>Importance of components:
                         PC1    PC2    PC3     PC4     PC5    PC6     PC7
Standard deviation     2.261 1.9126 1.5220 1.06476 0.80331 0.6382 0.61586
Proportion of Variance 0.365 0.2613 0.1655 0.08098 0.04609 0.0291 0.02709
Cumulative Proportion  0.365 0.6263 0.7918 0.87276 0.91885 0.9479 0.97504
                          PC8     PC9    PC10      PC11      PC12      PC13
Standard deviation     0.4908 0.25679 0.20650 8.377e-16 3.111e-16 2.272e-16
Proportion of Variance 0.0172 0.00471 0.00305 0.000e+00 0.000e+00 0.000e+00
Cumulative Proportion  0.9922 0.99695 1.00000 1.000e+00 1.000e+00 1.000e+00
                            PC14
Standard deviation     1.819e-16
Proportion of Variance 0.000e+00
Cumulative Proportion  1.000e+00</code></pre>
<pre class="r"><code># Compute a correlation matrix
cor.matrix &lt;- cor(df_saliva_scaled, use = &quot;pairwise.complete.obs&quot;)

corrplot(cor.matrix, type = &quot;upper&quot;, order = &quot;hclust&quot;, addCoef.col =&#39;black&#39;, number.cex = 0.8,
         tl.col = &quot;black&quot;, tl.srt = 45, tl.cex = .7, insig = &quot;blank&quot;)</code></pre>
<p><img src="figure/analysisv2.Rmd/cor-matrix-na-omit-1.png" width="1056" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cor-matrix-na-omit-1">
Past versions of cor-matrix-na-omit-1.png
</button>
</p>
<div id="fig-cor-matrix-na-omit-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/b5db62c16564a512ad4937891c3ee62e7802e59e/docs/figure/analysisv2.Rmd/cor-matrix-na-omit-1.png" target="_blank">b5db62c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/170c8ef65fdd1fd4c40944e49aab0563198a7516/docs/figure/analysisv2.Rmd/cor-matrix-na-omit-1.png" target="_blank">170c8ef</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Scree plot of variances
fviz_eig(pca)</code></pre>
<p><img src="figure/analysisv2.Rmd/scree-na-omit-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-scree-na-omit-1">
Past versions of scree-na-omit-1.png
</button>
</p>
<div id="fig-scree-na-omit-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/170c8ef65fdd1fd4c40944e49aab0563198a7516/docs/figure/analysisv2.Rmd/scree-na-omit-1.png" target="_blank">170c8ef</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Plot the variable factors 
fviz_pca_var(pca, col.var=&quot;contrib&quot;, 
             gradient.cols = c(&quot;#00AFBB&quot;, &quot;#E7B800&quot;, &quot;#FC4E07&quot;),
             repel = TRUE)</code></pre>
<p><img src="figure/analysisv2.Rmd/contrib-na-omit-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-contrib-na-omit-1">
Past versions of contrib-na-omit-1.png
</button>
</p>
<div id="fig-contrib-na-omit-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/170c8ef65fdd1fd4c40944e49aab0563198a7516/docs/figure/analysisv2.Rmd/contrib-na-omit-1.png" target="_blank">170c8ef</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="na-replaced-with-mean" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> NA replaced with
mean</h2>
<p>We repeat the analysis with means replacing missing values.</p>
<pre class="r"><code># Scale the Data
df_saliva_scaled &lt;- df_saliva %&gt;%
  select(all_of(vars_include)) %&gt;% 
  select(-&quot;Sample_ID&quot;) %&gt;% 
  # na.omit() %&gt;% 
  mutate(across(everything(), ~ replace(., is.na(.), mean(., na.rm = TRUE)))) %&gt;%
    mutate(across(everything(), ~ as.numeric(.) %&gt;% scale))


# Perform PCA
pca &lt;- df_saliva_scaled %&gt;%
  prcomp(center = FALSE, scale. = FALSE)

# Print the summary of the PCA object
summary(pca)</code></pre>
<pre><code>Importance of components:
                          PC1    PC2    PC3    PC4    PC5     PC6     PC7
Standard deviation     2.3221 1.8284 1.4599 1.0523 0.8060 0.72630 0.67157
Proportion of Variance 0.3851 0.2388 0.1522 0.0791 0.0464 0.03768 0.03221
Cumulative Proportion  0.3851 0.6239 0.7762 0.8553 0.9017 0.93937 0.97158
                           PC8     PC9    PC10    PC11      PC12      PC13
Standard deviation     0.49321 0.26609 0.24084 0.16058 3.546e-16 2.775e-16
Proportion of Variance 0.01738 0.00506 0.00414 0.00184 0.000e+00 0.000e+00
Cumulative Proportion  0.98896 0.99401 0.99816 1.00000 1.000e+00 1.000e+00
                            PC14
Standard deviation     2.037e-16
Proportion of Variance 0.000e+00
Cumulative Proportion  1.000e+00</code></pre>
<pre class="r"><code># Compute a correlation matrix
cor.matrix &lt;- cor(df_saliva_scaled, use = &quot;pairwise.complete.obs&quot;)

corrplot(cor.matrix, type = &quot;upper&quot;, order = &quot;hclust&quot;, addCoef.col =&#39;black&#39;, number.cex = 0.8,
         tl.col = &quot;black&quot;, tl.srt = 45, tl.cex = .7, insig = &quot;blank&quot;)</code></pre>
<p><img src="figure/analysisv2.Rmd/cor-matrix-na-mean-1.png" width="1056" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cor-matrix-na-mean-1">
Past versions of cor-matrix-na-mean-1.png
</button>
</p>
<div id="fig-cor-matrix-na-mean-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/b5db62c16564a512ad4937891c3ee62e7802e59e/docs/figure/analysisv2.Rmd/cor-matrix-na-mean-1.png" target="_blank">b5db62c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/170c8ef65fdd1fd4c40944e49aab0563198a7516/docs/figure/analysisv2.Rmd/cor-matrix-na-mean-1.png" target="_blank">170c8ef</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Scree plot of variances
fviz_eig(pca)</code></pre>
<p><img src="figure/analysisv2.Rmd/scree-na-mean-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-scree-na-mean-1">
Past versions of scree-na-mean-1.png
</button>
</p>
<div id="fig-scree-na-mean-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/170c8ef65fdd1fd4c40944e49aab0563198a7516/docs/figure/analysisv2.Rmd/scree-na-mean-1.png" target="_blank">170c8ef</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Plot the variable factors 
fviz_pca_var(pca, col.var=&quot;contrib&quot;, 
             gradient.cols = c(&quot;#00AFBB&quot;, &quot;#E7B800&quot;, &quot;#FC4E07&quot;),
             repel = TRUE)</code></pre>
<p><img src="figure/analysisv2.Rmd/contrib-na-mean-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-contrib-na-mean-1">
Past versions of contrib-na-mean-1.png
</button>
</p>
<div id="fig-contrib-na-mean-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/170c8ef65fdd1fd4c40944e49aab0563198a7516/docs/figure/analysisv2.Rmd/contrib-na-mean-1.png" target="_blank">170c8ef</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="reduced-variable-pca" class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Reduced variable
PCA</h2>
<p>In both cases, it appears as though it would be sufficient to take
the following variables as they are highly correlated in the same
directions (or a summary value representing both) And we will add two
combined flowrate sAA values based on the sAA after and before, and
change.</p>
<pre class="r"><code># Let&#39;s ensure ID column is a factor
df_saliva$Sample_ID &lt;- as.factor(df_saliva$Sample_ID)

# Save your non-numeric variable
df_label &lt;- df_saliva$Sample_ID
starch_freq &lt;- df_saliva$Starch_Frequency
citrus_length &lt;- df_saliva$Citrus_Sum_120sec

df_saliva_final &lt;- df_saliva %&gt;% 
  mutate(Flow_Rate_mean = (Flow_Rate_1_g_min + Flow_Rate_2_g_min)/2,
         sAA_flow_After = sAA_After*Flow_Rate_mean,
         sAA_flow_Before = sAA_Before*Flow_Rate_mean,
         sAA_flow_Change = sAA_Change*Flow_Rate_mean) %&gt;% 
  select(TP_Before, TP_After, Primer1_Diploid, Primer2_Diploid, TP_Change, sAA_Change, sAA_Before, sAA_After, Flow_Rate_mean, sAA_flow_Before, sAA_flow_After, sAA_flow_Change, Starch_Frequency, Citrus_Sum_120sec)</code></pre>
<pre class="r"><code># Scale your numeric data and store original row names
  
scaled_df &lt;- df_saliva_final %&gt;% 
  na.omit() %&gt;% 
  mutate(across(everything(), ~ replace(., is.na(.), mean(., na.rm = TRUE)))) %&gt;%
  mutate(across(everything(), ~scale(.x)))  

# Save the original order of your data
row_order &lt;- row.names(scaled_df)</code></pre>
<pre class="r"><code># Perform PCA
pca_results &lt;- prcomp(scaled_df, center = FALSE, scale = FALSE)

# Print the summary of the PCA object
summary(pca_results)</code></pre>
<pre><code>Importance of components:
                          PC1    PC2    PC3     PC4    PC5     PC6     PC7
Standard deviation     2.3513 1.5503 1.2667 1.05352 0.9864 0.92254 0.80371
Proportion of Variance 0.3949 0.1717 0.1146 0.07928 0.0695 0.06079 0.04614
Cumulative Proportion  0.3949 0.5666 0.6812 0.76045 0.8299 0.89074 0.93688
                           PC8     PC9   PC10    PC11      PC12      PC13
Standard deviation     0.68703 0.59484 0.2083 0.11993 8.062e-16 2.519e-16
Proportion of Variance 0.03372 0.02527 0.0031 0.00103 0.000e+00 0.000e+00
Cumulative Proportion  0.97060 0.99587 0.9990 1.00000 1.000e+00 1.000e+00
                            PC14
Standard deviation     1.943e-16
Proportion of Variance 0.000e+00
Cumulative Proportion  1.000e+00</code></pre>
<pre class="r"><code># Add original row names back to the scores. 
rownames(pca_results$x) &lt;- row_order

# Add Starch_Frequency back as a factor 
df_final &lt;- data.frame(pca_results$x, Starch_Frequency = starch_freq[match(row_order, df_label)])</code></pre>
<pre class="r"><code># Plot the variable factors 
fviz_pca_var(pca_results, col.var=&quot;contrib&quot;, 
             gradient.cols = c(&quot;#00AFBB&quot;, &quot;#E7B800&quot;, &quot;#FC4E07&quot;),
             repel = TRUE)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/analysisv2.Rmd/contrib-na-mean-final-1.png" alt="The largest contributions seem to be coming from the sAA_Before/After and the combination of sAA and flow rate" width="672" />
<p class="caption">
The largest contributions seem to be coming from the sAA_Before/After
and the combination of sAA and flow rate
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-contrib-na-mean-final-1">
Past versions of contrib-na-mean-final-1.png
</button>
</p>
<div id="fig-contrib-na-mean-final-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/eb2e32f709faaf9e78b11c8e22c1e29ade2d49bc/docs/figure/analysisv2.Rmd/contrib-na-mean-final-1.png" target="_blank">eb2e32f</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="starch-frequency-pca" class="section level3" number="1.3.1">
<h3><span class="header-section-number">1.3.1</span> Starch Frequency
PCA</h3>
<pre class="r"><code>fviz_pca_ind(pca_results, 
             label = &quot;none&quot;, 
             habillage = df_final$Starch_Frequency, 
             addEllipses = TRUE,
             palette = &quot;jco&quot;,
             legend.title = &quot;Starch Frequency&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/analysisv2.Rmd/df-final-contrib-starch-1.png" alt="There is no appreciable pattern in the PCA relating to the frequencyof starch consumption among participants." width="672" />
<p class="caption">
There is no appreciable pattern in the PCA relating to the frequencyof
starch consumption among participants.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-df-final-contrib-starch-1">
Past versions of df-final-contrib-starch-1.png
</button>
</p>
<div id="fig-df-final-contrib-starch-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/eb2e32f709faaf9e78b11c8e22c1e29ade2d49bc/docs/figure/analysisv2.Rmd/df-final-contrib-starch-1.png" target="_blank">eb2e32f</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/170c8ef65fdd1fd4c40944e49aab0563198a7516/docs/figure/analysisv2.Rmd/df-final-contrib-starch-1.png" target="_blank">170c8ef</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cnv-gradient-pca" class="section level3" number="1.3.2">
<h3><span class="header-section-number">1.3.2</span> CNV gradient
PCA</h3>
<pre class="r"><code># Before adding `Primer2_Diploid` variable to our new dataframe, ensure it&#39;s numeric
df_final$Primer2_Diploid &lt;- as.numeric(df_saliva$Primer2_Diploid[match(row_order, df_label)])

# Plot the PCA scores
ggplot(df_final, aes(x = PC1, y = PC2)) + 
  geom_point(aes(color = Primer2_Diploid)) + 
  scale_color_gradient(low = &quot;blue&quot;, high = &quot;green&quot;) +
  theme_minimal() +
  labs(color = &quot;Primer2_Diploid levels&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/analysisv2.Rmd/fig-final-pca-gradient-1.png" alt="CNV (as captured by Primer 2, does not seem to follow any gradient in the first two PCs" width="672" />
<p class="caption">
CNV (as captured by Primer 2, does not seem to follow any gradient in
the first two PCs
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig-final-pca-gradient-1">
Past versions of fig-final-pca-gradient-1.png
</button>
</p>
<div id="fig-fig-final-pca-gradient-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/eb2e32f709faaf9e78b11c8e22c1e29ade2d49bc/docs/figure/analysisv2.Rmd/fig-final-pca-gradient-1.png" target="_blank">eb2e32f</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/170c8ef65fdd1fd4c40944e49aab0563198a7516/docs/figure/analysisv2.Rmd/fig-final-pca-gradient-1.png" target="_blank">170c8ef</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="citrus-gradient-pca" class="section level3" number="1.3.3">
<h3><span class="header-section-number">1.3.3</span> Citrus gradient
PCA</h3>
<pre class="r"><code># Before adding `Citrus_Sum_120sec` variable to our new dataframe, ensure it&#39;s numeric
df_final$Citrus_Sum_120sec &lt;- as.numeric(df_saliva$Citrus_Sum_120sec[match(row_order, df_label)])

# Plot the PCA scores
ggplot(df_final, aes(x = PC1, y = PC2)) + 
  geom_point(aes(color = Citrus_Sum_120sec)) + 
  scale_color_gradient(low = &quot;blue&quot;, high = &quot;green&quot;) +
  theme_minimal() +
  labs(color = &quot;Citrus Length&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/analysisv2.Rmd/fig-final-pca-gradient-citrus-1.png" alt="Similarly, citrus perception (by time) does not appear to be shaped by the first two PCSs" width="672" />
<p class="caption">
Similarly, citrus perception (by time) does not appear to be shaped by
the first two PCSs
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig-final-pca-gradient-citrus-1">
Past versions of fig-final-pca-gradient-citrus-1.png
</button>
</p>
<div id="fig-fig-final-pca-gradient-citrus-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/eb2e32f709faaf9e78b11c8e22c1e29ade2d49bc/docs/figure/analysisv2.Rmd/fig-final-pca-gradient-citrus-1.png" target="_blank">eb2e32f</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/170c8ef65fdd1fd4c40944e49aab0563198a7516/docs/figure/analysisv2.Rmd/fig-final-pca-gradient-citrus-1.png" target="_blank">170c8ef</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="correlations" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Correlations</h1>
<p>Below is a general set of scatterplots looking at correlations
between different variables.</p>
<pre class="r"><code># Select only numeric columns
df_numeric &lt;- df_saliva_final[, sapply(df_saliva_final, is.numeric)]

# Create the scatter plot matrix
ggpairs(df_numeric)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/analysisv2.Rmd/fig-scatter-matrix-pearson-1.png" alt="Pearson's correlation" width="1440" />
<p class="caption">
Pearson’s correlation
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig-scatter-matrix-pearson-1">
Past versions of fig-scatter-matrix-pearson-1.png
</button>
</p>
<div id="fig-fig-scatter-matrix-pearson-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/b5db62c16564a512ad4937891c3ee62e7802e59e/docs/figure/analysisv2.Rmd/fig-scatter-matrix-pearson-1.png" target="_blank">b5db62c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/eb2e32f709faaf9e78b11c8e22c1e29ade2d49bc/docs/figure/analysisv2.Rmd/fig-scatter-matrix-pearson-1.png" target="_blank">eb2e32f</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/6e019773d11bd25f82f5d91be75b101a2da374d8/docs/figure/analysisv2.Rmd/fig-scatter-matrix-pearson-1.png" target="_blank">6e01977</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/170c8ef65fdd1fd4c40944e49aab0563198a7516/docs/figure/analysisv2.Rmd/fig-scatter-matrix-pearson-1.png" target="_blank">170c8ef</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Compute a correlation matrix
cor.matrix &lt;- cor(df_numeric, use = &quot;pairwise.complete.obs&quot;)

corrplot(cor.matrix, type = &quot;upper&quot;, order = &quot;hclust&quot;, addCoef.col =&#39;black&#39;, number.cex = 0.8,
         tl.col = &quot;black&quot;, tl.srt = 45, tl.cex = .7, insig = &quot;blank&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/analysisv2.Rmd/cor-matrix-final-pearson-1.png" alt="Pearson's correlation" width="1056" />
<p class="caption">
Pearson’s correlation
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cor-matrix-final-pearson-1">
Past versions of cor-matrix-final-pearson-1.png
</button>
</p>
<div id="fig-cor-matrix-final-pearson-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/b5db62c16564a512ad4937891c3ee62e7802e59e/docs/figure/analysisv2.Rmd/cor-matrix-final-pearson-1.png" target="_blank">b5db62c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Create the scatter plot matrix using spearman correlation
ggpairs(df_numeric, upper = list(continuous = wrap(&quot;cor&quot;, method = &quot;spearman&quot;)))</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/analysisv2.Rmd/fig-scatter-matrix-spearman-1.png" alt="Spearman's rank correlation" width="1440" />
<p class="caption">
Spearman’s rank correlation
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig-scatter-matrix-spearman-1">
Past versions of fig-scatter-matrix-spearman-1.png
</button>
</p>
<div id="fig-fig-scatter-matrix-spearman-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/b5db62c16564a512ad4937891c3ee62e7802e59e/docs/figure/analysisv2.Rmd/fig-scatter-matrix-spearman-1.png" target="_blank">b5db62c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/eb2e32f709faaf9e78b11c8e22c1e29ade2d49bc/docs/figure/analysisv2.Rmd/fig-scatter-matrix-spearman-1.png" target="_blank">eb2e32f</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/6e019773d11bd25f82f5d91be75b101a2da374d8/docs/figure/analysisv2.Rmd/fig-scatter-matrix-spearman-1.png" target="_blank">6e01977</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/170c8ef65fdd1fd4c40944e49aab0563198a7516/docs/figure/analysisv2.Rmd/fig-scatter-matrix-spearman-1.png" target="_blank">170c8ef</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Compute a correlation matrix
cor.matrix &lt;- cor(df_numeric, use = &quot;pairwise.complete.obs&quot;, method = &quot;spearman&quot;)

corrplot(cor.matrix, type = &quot;upper&quot;, order = &quot;hclust&quot;, addCoef.col =&#39;black&#39;, number.cex = 0.8,
         tl.col = &quot;black&quot;, tl.srt = 45, tl.cex = .7, insig = &quot;blank&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/analysisv2.Rmd/cor-matrix-final-spearman-1.png" alt="Spearman's rank correlation with different visualization of correlations" width="1056" />
<p class="caption">
Spearman’s rank correlation with different visualization of correlations
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cor-matrix-final-spearman-1">
Past versions of cor-matrix-final-spearman-1.png
</button>
</p>
<div id="fig-cor-matrix-final-spearman-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/b5db62c16564a512ad4937891c3ee62e7802e59e/docs/figure/analysisv2.Rmd/cor-matrix-final-spearman-1.png" target="_blank">b5db62c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="relationship-between-cnv-protein-and-saa"
class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Relationship between
CNV, protein and sAA</h2>
<pre class="r"><code>df_saliva_lm &lt;- df_saliva %&gt;% 
  mutate(Flow_Rate_mean = (Flow_Rate_1_g_min + Flow_Rate_2_g_min)/2,
         sAA_flow_After = sAA_After*Flow_Rate_mean,
         sAA_flow_Before = sAA_Before*Flow_Rate_mean,
         sAA_flow_Change = sAA_Change*Flow_Rate_mean) %&gt;% 
  select(TP_Before, TP_After, Primer1_Diploid, Primer2_Diploid, TP_Change, sAA_Change, sAA_Before, sAA_After, Flow_Rate_mean, sAA_flow_Before, sAA_flow_After, sAA_flow_Change, Starch_Frequency, Citrus_Sum_120sec) %&gt;% 
  na.omit()</code></pre>
<pre class="r"><code># Generalized function to toggle different timepoints or changes
run_mediation_model_Primer2 &lt;- function(df, timepoint_or_change) {
    mediation_model &lt;- paste(&quot;
        # direct effect
        sAA_&quot;, timepoint_or_change, &quot; ~ c*Primer2_Diploid
        # mediator
        Flow_Rate_mean ~ a*Primer2_Diploid
        TP_&quot;, timepoint_or_change, &quot; ~ m*Primer2_Diploid
        sAA_&quot;, timepoint_or_change, &quot; ~ b*Flow_Rate_mean + n*TP_&quot;, timepoint_or_change, &quot;
        # indirect effect (a*b and m*n)
        indirect1 := a*b
        indirect2 := m*n
        # total effect
        total := c + (a*b) + (m*n)
     &quot;, sep=&quot;&quot;)

    # Fit the model
    fit &lt;- sem(mediation_model, data=df)

    # Return the model
    return(fit)
}

run_mediation_model_Primer1 &lt;- function(df, timepoint_or_change) {
    mediation_model &lt;- paste(&quot;
        # direct effect
        sAA_&quot;, timepoint_or_change, &quot; ~ c*Primer1_Diploid
        # mediator
        Flow_Rate_mean ~ a*Primer1_Diploid
        TP_&quot;, timepoint_or_change, &quot; ~ m*Primer1_Diploid
        sAA_&quot;, timepoint_or_change, &quot; ~ b*Flow_Rate_mean + n*TP_&quot;, timepoint_or_change, &quot;
        # indirect effect (a*b and m*n)
        indirect1 := a*b
        indirect2 := m*n
        # total effect
        total := c + (a*b) + (m*n)
     &quot;, sep=&quot;&quot;)

    # Fit the model
    fit &lt;- sem(mediation_model, data=df)

    # Return the model
    return(fit)
}

run_mediation_model_Primer1_sAA_flow_TP_mediator &lt;- function(df, timepoint_or_change) {
    mediation_model &lt;- paste(&quot;
        # direct effect
        sAA_flow_&quot;, timepoint_or_change, &quot; ~ c*Primer1_Diploid
        # mediator
        TP_&quot;, timepoint_or_change, &quot; ~ a*Primer1_Diploid
        sAA_flow_&quot;, timepoint_or_change, &quot; ~ b*TP_&quot;, timepoint_or_change, &quot;
        # indirect effect (a*b)
        indirect1 := a*b
        # total effect
        total := c + (a*b)
     &quot;, sep=&quot;&quot;)

    # Fit the model
    fit &lt;- sem(mediation_model, data=df)

    # Return the model
    return(fit)
}

run_mediation_model_Primer2_sAA_flow_TP_mediator &lt;- function(df, timepoint_or_change) {
    mediation_model &lt;- paste(&quot;
        # direct effect
        sAA_flow_&quot;, timepoint_or_change, &quot; ~ c*Primer2_Diploid
        # mediator
        TP_&quot;, timepoint_or_change, &quot; ~ a*Primer2_Diploid
        sAA_flow_&quot;, timepoint_or_change, &quot; ~ b*TP_&quot;, timepoint_or_change, &quot;
        # indirect effect (a*b)
        indirect1 := a*b
        # total effect
        total := c + (a*b)
     &quot;, sep=&quot;&quot;)

    # Fit the model
    fit &lt;- sem(mediation_model, data=df)

    # Return the model
    return(fit)
}



library(semPlot)

plot_sem_model &lt;- function(model) {
  # create the plot
  semPaths(model, 
           &quot;std&quot;, # standardized coefficients
           &quot;est&quot;, # show estimates
           curveAdjacent = TRUE, # curved lines for better distinction of paths
           style = &quot;lisrel&quot;) # style of the plot
}</code></pre>
<div id="cnv-to-saa-with-mediator-tp-and-flow-rate"
class="section level3" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> CNV to sAA with
mediator TP and Flow rate</h3>
<div id="before" class="section level4" number="2.1.1.1">
<h4><span class="header-section-number">2.1.1.1</span> Before</h4>
<pre class="r"><code># Run the mediation model for Primer1
fit_before_P1 &lt;- run_mediation_model_Primer1(df_saliva_lm, &#39;Before&#39;)
# Summarize the results
cat(&quot;Summary for Primer1 Model:\n&quot;)</code></pre>
<pre><code>Summary for Primer1 Model:</code></pre>
<pre class="r"><code>summary(fit_before_P1)</code></pre>
<pre><code>lavaan 0.6.15 ended normally after 16 iterations

  Estimator                                         ML
  Optimization method                           NLMINB
  Number of model parameters                         8

  Number of observations                            68

Model Test User Model:
                                                      
  Test statistic                                17.280
  Degrees of freedom                                 1
  P-value (Chi-square)                           0.000

Parameter Estimates:

  Standard errors                             Standard
  Information                                 Expected
  Information saturated (h1) model          Structured

Regressions:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
  sAA_Before ~                                           
    Prmr1_Dpld (c)     10.448     2.028    5.153    0.000
  Flow_Rate_mean ~                                       
    Prmr1_Dpld (a)      0.024     0.011    2.190    0.029
  TP_Before ~                                            
    Prmr1_Dpld (m)      0.939    13.873    0.068    0.946
  sAA_Before ~                                           
    Flow_Rt_mn (b)     22.802    22.053    1.034    0.301
    TP_Before  (n)      0.089     0.017    5.184    0.000

Variances:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
   .sAA_Before       5050.515   866.156    5.831    0.000
   .Flow_Rate_mean      0.153     0.026    5.831    0.000
   .TP_Before      253114.182 43408.724    5.831    0.000

Defined Parameters:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
    indirect1           0.538     0.576    0.935    0.350
    indirect2           0.083     1.232    0.068    0.946
    total              11.069     2.328    4.755    0.000</code></pre>
<p>Summary of path analysis:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Model Fit</strong>: The chi-square test of model fit is
significant (Chi-square = 17.280, df = 1, p = 0.000). This means that
the specified model does not fit the data well.</p></li>
<li><p><strong>Path Coefficients</strong>:</p>
<ul>
<li><strong><code>Prmr1_Dpld</code></strong> to
<strong><code>sAA_Before</code></strong> (path c): The coefficient is
positive and significant (Est.= 10.448, p = 0.000). This indicates that
<code>Prmr1_Dpld</code> has a positive direct effect on
<code>sAA_Before</code>.</li>
<li><strong><code>Prmr1_Dpld</code></strong> to
<strong><code>Flow_Rate_mean</code></strong> (path a): The coefficient
is positive and significant (Est.= 0.024, p = 0.029), indicating that
<code>Prmr1_Dpld</code> positively influences
<code>Flow_Rate_mean</code>.</li>
<li><strong><code>TP_Before</code></strong> to
<strong><code>sAA_Before</code></strong> (path n): This path is also
significant (Est.= 0.089, p = 0.000), suggesting that
<code>TP_Before</code> has a positive and significant effect on
<code>sAA_Before</code>.</li>
<li>other paths (m, b) are not statistically significant.</li>
</ul></li>
<li><p><strong>Variances</strong>: All the residual variances of the
endogenous variables (<code>.sAA_Before</code>,
<code>.Flow_Rate_mean</code>, <code>.TP_Before</code>) are significantly
different from zero.</p></li>
<li><p><strong>Mediation Analysis</strong>: Both indirect effects
through <code>Flow_Rate_mean</code> (indirect1) and
<code>TP_Before</code> (indirect2) were not statistically significant
(p-values = 0.350 and 0.946 respectively). This suggests that there is
no significant mediating effect of <code>Flow_Rate_mean</code> and
<code>TP_Before</code> in the relationship between
<code>Prmr1_Dpld</code> and <code>sAA_Before</code>.</p></li>
<li><p><strong>Total effect</strong>: The total effect of
<code>Prmr1_Dpld</code> on <code>sAA_Before</code> is significant and
positive (Est.= 11.069, p = 0.000), suggesting <code>Prmr1_Dpld</code>
has a significant direct impact on <code>sAA_Before</code>.</p></li>
</ol>
<p>In conclusion, <code>Prmr1_Dpld</code> has a significant direct
influence on <code>sAA_Before</code> and <code>Flow_Rate_mean</code>.
Additionally, <code>TP_Before</code> significantly affects
<code>sAA_Before</code>. However, neither <code>Flow_Rate_mean</code>
nor <code>TP_Before</code> significantly mediates the effect of
<code>Prmr1_Dpld</code> on <code>sAA_Before</code>.</p>
<pre class="r"><code># Run the mediation model for Primer2
fit_before_P2 &lt;- run_mediation_model_Primer2(df_saliva_lm, &#39;Before&#39;)
# Summarize the results
cat(&quot;\n\nSummary for Primer2 Model:\n&quot;)</code></pre>
<pre><code>

Summary for Primer2 Model:</code></pre>
<pre class="r"><code>summary(fit_before_P2)</code></pre>
<pre><code>lavaan 0.6.15 ended normally after 1 iteration

  Estimator                                         ML
  Optimization method                           NLMINB
  Number of model parameters                         8

  Number of observations                            68

Model Test User Model:
                                                      
  Test statistic                                14.539
  Degrees of freedom                                 1
  P-value (Chi-square)                           0.000

Parameter Estimates:

  Standard errors                             Standard
  Information                                 Expected
  Information saturated (h1) model          Structured

Regressions:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
  sAA_Before ~                                           
    Prmr2_Dpld (c)     12.276     2.531    4.850    0.000
  Flow_Rate_mean ~                                       
    Prmr2_Dpld (a)      0.027     0.013    2.040    0.041
  TP_Before ~                                            
    Prmr2_Dpld (m)    -19.974    16.867   -1.184    0.236
  sAA_Before ~                                           
    Flow_Rt_mn (b)     36.084    22.199    1.625    0.104
    TP_Before  (n)      0.107     0.017    6.106    0.000

Variances:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
   .sAA_Before       5162.654   885.388    5.831    0.000
   .Flow_Rate_mean      0.154     0.026    5.831    0.000
   .TP_Before      248016.549 42534.487    5.831    0.000

Defined Parameters:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
    indirect1           0.979     0.770    1.271    0.204
    indirect2          -2.134     1.836   -1.163    0.245
    total              11.121     3.066    3.627    0.000</code></pre>
<p>Here’s a summary of the notable results:</p>
<ul>
<li><p><strong>Direct Effect</strong>: A significant direct effect
exists between <strong><code>Prmr2_Dpld</code></strong> and
<strong><code>sAA_Before</code></strong>. This means that when adjusting
for other variables, any increase in <code>Prmr2_Dpld</code> will
typically result in an increase in <code>sAA_Before</code>.</p></li>
<li><p><strong>Other Significant Path</strong>: A significant connection
also exists between <strong><code>Prmr2_Dpld</code></strong> and
<strong><code>Flow_Rate_mean</code></strong>. Furthermore,
<strong><code>TP_Before</code></strong> also exhibits a definite
influence on <strong><code>sAA_Before</code></strong>.</p></li>
<li><p><strong>Total Effect</strong>: The total impact of
<strong><code>Prmr2_Dpld</code></strong> on
<strong><code>sAA_Before</code></strong> is positive and significant,
establishing that <code>Prmr2_Dpld</code> has a dominating influence on
<code>sAA_Before</code>.</p></li>
<li><p><strong>Mediating Effect</strong>: Despite the numerous
significant paths and a total effect, there is <strong>no evidence of a
significant mediating effect</strong>. Neither
<code>Flow_Rate_mean</code> nor <code>TP_Before</code> effectively
mediate the connection between <code>Prmr2_Dpld</code> and
<code>sAA_Before</code>. This means that the influence exerted by
<code>Prmr2_Dpld</code> on <code>sAA_Before</code> is not significantly
conveyed through these two variables.</p></li>
</ul>
<p>In conclusion, the <strong>best model</strong> from these results
underscores direct paths between <code>Prmr2_Dpld</code> and
<code>sAA_Before</code> and between <code>Prmr2_Dpld</code> and
<code>Flow_Rate_mean</code>, as well as between <code>TP_Before</code>
and <code>sAA_Before</code>. While <code>Flow_Rate_mean</code> and
<code>TP_Before</code> may be interesting to explore, they do not play
significant roles as mediators in this model, and thus might not greatly
enhance our ability to predict <code>sAA_Before</code>.</p>
<pre class="r"><code>layout(matrix(1:2, nrow = 1, byrow = TRUE))
# Tags/labels for each plot
layout_title &lt;- c(&quot;Primer 1&quot;, &quot;Primer 2&quot;)

# Create the first plot
plot_sem_model(fit_before_P1)
title(layout_title[1])

# Create the second plot
plot_sem_model(fit_before_P2)
title(layout_title[2])</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/analysisv2.Rmd/fig-med-before-1.png" alt="Mediator models for values before experiment" width="672" />
<p class="caption">
Mediator models for values before experiment
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig-med-before-1">
Past versions of fig-med-before-1.png
</button>
</p>
<div id="fig-fig-med-before-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/eb2e32f709faaf9e78b11c8e22c1e29ade2d49bc/docs/figure/analysisv2.Rmd/fig-med-before-1.png" target="_blank">eb2e32f</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/170c8ef65fdd1fd4c40944e49aab0563198a7516/docs/figure/analysisv2.Rmd/fig-med-before-1.png" target="_blank">170c8ef</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="after" class="section level4" number="2.1.1.2">
<h4><span class="header-section-number">2.1.1.2</span> After</h4>
<pre class="r"><code># Run the mediation model for Primer1
fit_after_P1 &lt;- run_mediation_model_Primer1(df_saliva_lm, &#39;After&#39;)
# Summarize the results
cat(&quot;Summary for Primer1 Model:\n&quot;)</code></pre>
<pre><code>Summary for Primer1 Model:</code></pre>
<pre class="r"><code>summary(fit_after_P1)</code></pre>
<pre><code>lavaan 0.6.15 ended normally after 1 iteration

  Estimator                                         ML
  Optimization method                           NLMINB
  Number of model parameters                         8

  Number of observations                            68

Model Test User Model:
                                                      
  Test statistic                                17.230
  Degrees of freedom                                 1
  P-value (Chi-square)                           0.000

Parameter Estimates:

  Standard errors                             Standard
  Information                                 Expected
  Information saturated (h1) model          Structured

Regressions:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
  sAA_After ~                                            
    Prmr1_Dpld (c)      6.368     1.267    5.025    0.000
  Flow_Rate_mean ~                                       
    Prmr1_Dpld (a)      0.024     0.011    2.190    0.029
  TP_After ~                                             
    Prmr1_Dpld (m)     14.561     9.854    1.478    0.139
  sAA_After ~                                            
    Flow_Rt_mn (b)    -16.666    13.581   -1.227    0.220
    TP_After   (n)      0.001     0.015    0.083    0.934

Variances:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
   .sAA_After        1915.261   328.465    5.831    0.000
   .Flow_Rate_mean      0.153     0.026    5.831    0.000
   .TP_After       127687.632 21898.248    5.831    0.000

Defined Parameters:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
    indirect1          -0.393     0.367   -1.071    0.284
    indirect2           0.018     0.217    0.083    0.934
    total               5.993     1.220    4.911    0.000</code></pre>
<p>Here is a summary of the results for Primer 1:</p>
<ul>
<li><p><strong>Direct Effect</strong>: A significant direct effect
exists between <strong><code>Prmr1_Dpld</code></strong> and
<strong><code>sAA_After</code></strong>. This suggests that an increase
in <code>Prmr1_Dpld</code> is associated with an increase in
<code>sAA_After</code> while controlling for other factors.</p></li>
<li><p><strong>Other Significant Relationships</strong>: The only other
significant effect is between <strong><code>Prmr1_Dpld</code></strong>
and <strong><code>Flow_Rate_mean</code></strong>.</p></li>
<li><p><strong>Total Effect</strong>: The total impact of
<strong><code>Prmr1_Dpld</code></strong> on
<strong><code>sAA_After</code></strong> is positive and significant,
meaning that <code>Prmr1_Dpld</code> substantially influences
<code>sAA_After</code>.</p></li>
<li><p><strong>Mediating Effect</strong>: Neither
<strong><code>Flow_Rate_mean</code></strong> nor
<strong><code>TP_After</code></strong> show a significant mediating
effect in the relationship between <code>Prmr1_Dpld</code> and
<code>sAA_After</code>. This means that the influence of
<code>Prmr1_Dpld</code> on <code>sAA_After</code> is not significantly
transferred through these two variables.</p></li>
</ul>
<p>Thus, the <strong>best model</strong> from these results includes
direct paths between <code>Prmr1_Dpld</code> and <code>sAA_After</code>,
and between <code>Prmr1_Dpld</code> and <code>Flow_Rate_mean</code>.
Similar to previous models, <code>Flow_Rate_mean</code> and
<code>TP_After</code> may be interesting to explore but they do not make
significant contributions as mediators in the model, and thus may not
improve our predictive capability for <code>sAA_After</code>.</p>
<pre class="r"><code># Run the mediation model for Primer2
fit_after_P2 &lt;- run_mediation_model_Primer2(df_saliva_lm, &#39;After&#39;)
# Summarize the results
cat(&quot;\n\nSummary for Primer2 Model:\n&quot;)</code></pre>
<pre><code>

Summary for Primer2 Model:</code></pre>
<pre class="r"><code>summary(fit_after_P2)</code></pre>
<pre><code>lavaan 0.6.15 ended normally after 17 iterations

  Estimator                                         ML
  Optimization method                           NLMINB
  Number of model parameters                         8

  Number of observations                            68

Model Test User Model:
                                                      
  Test statistic                                13.803
  Degrees of freedom                                 1
  P-value (Chi-square)                           0.000

Parameter Estimates:

  Standard errors                             Standard
  Information                                 Expected
  Information saturated (h1) model          Structured

Regressions:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
  sAA_After ~                                            
    Prmr2_Dpld (c)      6.568     1.588    4.137    0.000
  Flow_Rate_mean ~                                       
    Prmr2_Dpld (a)      0.027     0.013    2.040    0.041
  TP_After ~                                             
    Prmr2_Dpld (m)      4.573    12.283    0.372    0.710
  sAA_After ~                                            
    Flow_Rt_mn (b)     -7.465    14.045   -0.532    0.595
    TP_After   (n)      0.016     0.015    1.057    0.291

Variances:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
   .sAA_After        2066.489   354.400    5.831    0.000
   .Flow_Rate_mean      0.154     0.026    5.831    0.000
   .TP_After       131519.420 22555.395    5.831    0.000

Defined Parameters:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
    indirect1          -0.202     0.394   -0.514    0.607
    indirect2           0.073     0.209    0.351    0.725
    total               6.438     1.555    4.140    0.000</code></pre>
<p>Here’s a summary of the findings from the model for Primer 2:</p>
<ul>
<li><p><strong>Direct Effect</strong>: There’s a significant direct
effect between <strong><code>Prmr2_Dpld</code></strong> and
<strong><code>sAA_After</code></strong>. Hence, in principle, increasing
<code>Prmr2_Dpld</code> results in an increase in <code>sAA_After</code>
when we adjust for other factors.</p></li>
<li><p><strong>Other Significant Relationships</strong>: The only
additional significant relationship in this model is between
<strong><code>Prmr2_Dpld</code></strong> and
<strong><code>Flow_Rate_mean</code></strong>.</p></li>
<li><p><strong>Total Effect</strong>: The total influence of
<strong><code>Prmr2_Dpld</code></strong> on
<strong><code>sAA_After</code></strong> is positive and significant,
showing that <code>Prmr2_Dpld</code> has a significant overall influence
on <code>sAA_After</code>.</p></li>
<li><p><strong>Mediating Effect</strong>: Neither
<strong><code>Flow_Rate_mean</code></strong> nor
<strong><code>TP_After</code></strong> demonstrates a significant
mediation of the relationship between <code>Prmr2_Dpld</code> and
<code>sAA_After</code>. Therefore, the effect of <code>Prmr2_Dpld</code>
on <code>sAA_After</code> is not significantly transmitted through these
two potential mediators.</p></li>
</ul>
<p>As a result, the <strong>best model</strong> developed from these
results includes direct paths between <code>Prmr2_Dpld</code> and
<code>sAA_After</code>, as well as between <code>Prmr2_Dpld</code> and
<code>Flow_Rate_mean</code>. Again, while <code>Flow_Rate_mean</code>
and <code>TP_After</code> might be intriguing to explore, they do not
make significant contributions to the model as mediators and thus might
not improve our predictive abilities concerning
<code>sAA_After</code>.</p>
<pre class="r"><code># Set up layout
layout(matrix(1:2, nrow = 1, byrow = TRUE))
# Tags/labels for each plot
layout_title &lt;- c(&quot;Primer 1&quot;, &quot;Primer 2&quot;)

# Create the first plot
plot_sem_model(fit_after_P1)
title(layout_title[1])

# Create the second plot
plot_sem_model(fit_after_P2)
title(layout_title[2])</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/analysisv2.Rmd/fig-med-after-1.png" alt="Mediator model for values after experiment" width="672" />
<p class="caption">
Mediator model for values after experiment
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig-med-after-1">
Past versions of fig-med-after-1.png
</button>
</p>
<div id="fig-fig-med-after-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/eb2e32f709faaf9e78b11c8e22c1e29ade2d49bc/docs/figure/analysisv2.Rmd/fig-med-after-1.png" target="_blank">eb2e32f</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/170c8ef65fdd1fd4c40944e49aab0563198a7516/docs/figure/analysisv2.Rmd/fig-med-after-1.png" target="_blank">170c8ef</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="change" class="section level4" number="2.1.1.3">
<h4><span class="header-section-number">2.1.1.3</span> Change</h4>
<pre class="r"><code># Run the mediation model for Primer1
fit_change_P1 &lt;- run_mediation_model_Primer1(df_saliva_lm, &#39;Change&#39;)
# Summarize the results
cat(&quot;Summary for Primer1 Model:\n&quot;)</code></pre>
<pre><code>Summary for Primer1 Model:</code></pre>
<pre class="r"><code>summary(fit_change_P1)</code></pre>
<pre><code>lavaan 0.6.15 ended normally after 1 iteration

  Estimator                                         ML
  Optimization method                           NLMINB
  Number of model parameters                         8

  Number of observations                            68

Model Test User Model:
                                                      
  Test statistic                                 1.302
  Degrees of freedom                                 1
  P-value (Chi-square)                           0.254

Parameter Estimates:

  Standard errors                             Standard
  Information                                 Expected
  Information saturated (h1) model          Structured

Regressions:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
  sAA_Change ~                                           
    Prmr1_Dpld (c)     -5.739     1.397   -4.109    0.000
  Flow_Rate_mean ~                                       
    Prmr1_Dpld (a)      0.024     0.011    2.190    0.029
  TP_Change ~                                            
    Prmr1_Dpld (m)     13.622    13.870    0.982    0.326
  sAA_Change ~                                           
    Flow_Rt_mn (b)      7.993    15.092    0.530    0.596
    TP_Change  (n)      0.035     0.012    2.966    0.003

Variances:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
   .sAA_Change       2365.395   405.662    5.831    0.000
   .Flow_Rate_mean      0.153     0.026    5.831    0.000
   .TP_Change      253012.389 43391.267    5.831    0.000

Defined Parameters:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
    indirect1           0.189     0.366    0.515    0.607
    indirect2           0.474     0.508    0.932    0.351
    total              -5.077     1.428   -3.555    0.000</code></pre>
<p>The model provides a good fit to the data (Chi-square = 1.302, p =
0.254).</p>
<ul>
<li>There is a significant negative direct effect of
<strong><code>Prmr1_Dpld</code></strong> on
<strong><code>sAA_Change</code></strong> (Est = -5.739, p = 0.000).</li>
<li>Furthermore, <strong><code>Prmr1_Dpld</code></strong> also has a
positive and significant effect on
<strong><code>Flow_Rate_mean</code></strong> (Est = 0.024, p =
0.029).</li>
<li>Lastly, <strong><code>TP_Change</code></strong> significantly
influences <strong><code>sAA_Change</code></strong> (Est = 0.035, p =
0.003).</li>
</ul>
<p>However, there’s no evidence of any significant mediating effects:
neither <strong><code>Flow_Rate_mean</code></strong> (Est = 0.189, p =
0.607) nor <strong><code>TP_Change</code></strong> (Est = 0.474, p =
0.351) mediates the relationship between <code>Prmr1_Dpld</code> and
<code>sAA_Change</code>.</p>
<p>The overall effect of <code>Prmr1_Dpld</code> on
<code>sAA_Change</code> is significant and negative (Est = -5.077, p =
0.000).</p>
<p>In conclusion, the <strong>best model</strong> includes direct paths
from <code>Prmr1_Dpld</code> to <code>sAA_Change</code>,
<code>Prmr1_Dpld</code> to <code>Flow_Rate_mean</code>, and
<code>TP_Change</code> to <code>sAA_Change</code>.
<code>Flow_Rate_mean</code> and <code>TP_Change</code> do not
significantly mediate the effect of <code>Prmr1_Dpld</code> on
<code>sAA_Change</code>.</p>
<pre class="r"><code># Run the mediation model for Primer2
fit_change_P2 &lt;- run_mediation_model_Primer2(df_saliva_lm, &#39;Change&#39;)
# Summarize the results
cat(&quot;\n\nSummary for Primer2 Model:\n&quot;)</code></pre>
<pre><code>

Summary for Primer2 Model:</code></pre>
<pre class="r"><code>summary(fit_change_P2)</code></pre>
<pre><code>lavaan 0.6.15 ended normally after 1 iteration

  Estimator                                         ML
  Optimization method                           NLMINB
  Number of model parameters                         8

  Number of observations                            68

Model Test User Model:
                                                      
  Test statistic                                 1.095
  Degrees of freedom                                 1
  P-value (Chi-square)                           0.295

Parameter Estimates:

  Standard errors                             Standard
  Information                                 Expected
  Information saturated (h1) model          Structured

Regressions:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
  sAA_Change ~                                           
    Prmr2_Dpld (c)     -5.677     1.796   -3.161    0.002
  Flow_Rate_mean ~                                       
    Prmr2_Dpld (a)      0.027     0.013    2.040    0.041
  TP_Change ~                                            
    Prmr2_Dpld (m)     24.547    16.896    1.453    0.146
  sAA_Change ~                                           
    Flow_Rt_mn (b)      3.715    15.678    0.237    0.813
    TP_Change  (n)      0.036     0.012    2.952    0.003

Variances:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
   .sAA_Change       2574.912   441.594    5.831    0.000
   .Flow_Rate_mean      0.154     0.026    5.831    0.000
   .TP_Change      248875.953 42681.874    5.831    0.000

Defined Parameters:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
    indirect1           0.101     0.428    0.235    0.814
    indirect2           0.894     0.686    1.303    0.192
    total              -4.683     1.826   -2.564    0.010</code></pre>
<p>The model fits the data well (Chi-square = 1.095, p = 0.295).</p>
<ul>
<li>There is a significant negative direct effect of
<strong><code>Prmr2_Dpld</code></strong> on
<strong><code>sAA_Change</code></strong> (Est = -5.677, p = 0.002).</li>
<li>Moreover, <strong><code>Prmr2_Dpld</code></strong> has a positive
significant effect on <strong><code>Flow_Rate_mean</code></strong> (Est
= 0.027, p = 0.041).</li>
<li><strong><code>TP_Change</code></strong> also significantly
influences <strong><code>sAA_Change</code></strong> (Est = 0.036, p =
0.003).</li>
</ul>
<p>However, neither <strong><code>Flow_Rate_mean</code></strong> (Est =
0.101, p = 0.814) nor <strong><code>TP_Change</code></strong> (Est =
0.894, p = 0.192) provides a significant mediating effect between
<code>Prmr2_Dpld</code> and <code>sAA_Change</code>.</p>
<p>The total effect of <code>Prmr2_Dpld</code> on
<code>sAA_Change</code> is significant and negative (Est = -4.683, p =
0.010).</p>
<p>In conclusion, the <strong>best model</strong> includes direct paths
from <code>Prmr2_Dpld</code> to <code>sAA_Change</code>,
<code>Prmr2_Dpld</code> to <code>Flow_Rate_mean</code>, and
<code>TP_Change</code> to <code>sAA_Change</code>. While
<code>Flow_Rate_mean</code> and <code>TP_Change</code> are included in
the model, there is no significant evidence to suggest that they mediate
the effect of <code>Prmr2_Dpld</code> on <code>sAA_Change</code>.</p>
<pre class="r"><code># set up layout 
layout(matrix(1:2, nrow = 1, byrow = TRUE))
# Tags/labels for each plot
layout_title &lt;- c(&quot;Primer 1&quot;, &quot;Primer 2&quot;)

# Create the first plot
plot_sem_model(fit_change_P1)
title(layout_title[1])

# Create the second plot
plot_sem_model(fit_change_P2)
title(layout_title[2])</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/analysisv2.Rmd/fig-med-change-1.png" alt="Mediator model for values changed during experiment" width="672" />
<p class="caption">
Mediator model for values changed during experiment
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig-med-change-1">
Past versions of fig-med-change-1.png
</button>
</p>
<div id="fig-fig-med-change-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/eb2e32f709faaf9e78b11c8e22c1e29ade2d49bc/docs/figure/analysisv2.Rmd/fig-med-change-1.png" target="_blank">eb2e32f</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/170c8ef65fdd1fd4c40944e49aab0563198a7516/docs/figure/analysisv2.Rmd/fig-med-change-1.png" target="_blank">170c8ef</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="cnv-to-saaflow-with-tp-mediator" class="section level3"
number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> CNV to sAA*flow
with TP mediator</h3>
<div id="before-1" class="section level4" number="2.1.2.1">
<h4><span class="header-section-number">2.1.2.1</span> Before</h4>
<pre class="r"><code># Run the mediation model for Primer1
fit_before_sAA_flow_Primer1 &lt;- run_mediation_model_Primer1_sAA_flow_TP_mediator(df_saliva_lm, &#39;Before&#39;)
# Summarize the results
cat(&quot;Summary for Primer1 Model:\n&quot;)</code></pre>
<pre><code>Summary for Primer1 Model:</code></pre>
<pre class="r"><code>summary(fit_before_sAA_flow_Primer1)</code></pre>
<pre><code>lavaan 0.6.15 ended normally after 10 iterations

  Estimator                                         ML
  Optimization method                           NLMINB
  Number of model parameters                         5

  Number of observations                            68

Model Test User Model:
                                                      
  Test statistic                                 0.000
  Degrees of freedom                                 0

Parameter Estimates:

  Standard errors                             Standard
  Information                                 Expected
  Information saturated (h1) model          Structured

Regressions:
                    Estimate    Std.Err   z-value  P(&gt;|z|)
  sAA_flow_Before ~                                       
    Prmr1_Dpld (c)      13.386     2.587    5.174    0.000
  TP_Before ~                                             
    Prmr1_Dpld (a)       0.939    13.873    0.068    0.946
  sAA_flow_Before ~                                       
    TP_Before  (b)       0.037     0.023    1.616    0.106

Variances:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
   .sAA_flow_Befor   8801.515  1509.447    5.831    0.000
   .TP_Before      253114.316 43408.747    5.831    0.000

Defined Parameters:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
    indirect1           0.034     0.507    0.068    0.946
    total              13.420     2.636    5.091    0.000</code></pre>
<p>The model fits the data perfectly when considering the provided
indicators (Chi-square = 0.000, degrees of freedom = 0).</p>
<ul>
<li><p>There is a significant positive direct effect of
<strong><code>Prmr1_Dpld</code></strong> on
<strong><code>sAA_flow_Before</code></strong> (Est = 13.386, p =
0.000).</p></li>
<li><p><strong><code>Prmr1_Dpld</code></strong> does not significantly
affect <strong><code>TP_Before</code></strong> (Est = 0.939, p = 0.946),
and the effect of <strong><code>TP_Before</code></strong> on
<strong><code>sAA_flow_Before</code></strong> is also not significant
(Est = 0.037, p = 0.106).</p></li>
<li><p>There is no significant indirect effect via
<strong><code>TP_Before</code></strong> (indirect1) (Est = 0.034, p =
0.946) between <code>Prmr1_Dpld</code> and
<code>sAA_flow_Before</code>.</p></li>
<li><p>The total effect of <code>Prmr1_Dpld</code> on
<code>sAA_flow_Before</code> is significant and positive (Est = 13.420,
p = 0.000).</p></li>
</ul>
<p>In conclusion, the <strong>best model</strong> includes a direct path
from <code>Prmr1_Dpld</code> to <code>sAA_flow_Before</code>. However,
the model shows no significant evidence that <code>TP_Before</code>
mediates the effect of <code>Prmr1_Dpld</code> on
<code>sAA_flow_Before</code>.</p>
<pre class="r"><code># Run the mediation model for Primer2
fit_before_sAA_flow_Primer2 &lt;- run_mediation_model_Primer2_sAA_flow_TP_mediator(df_saliva_lm, &#39;Before&#39;)
# Summarize the results
cat(&quot;\n\nSummary for Primer2 Model:\n&quot;)</code></pre>
<pre><code>

Summary for Primer2 Model:</code></pre>
<pre class="r"><code>summary(fit_before_sAA_flow_Primer2)</code></pre>
<pre><code>lavaan 0.6.15 ended normally after 1 iteration

  Estimator                                         ML
  Optimization method                           NLMINB
  Number of model parameters                         5

  Number of observations                            68

Model Test User Model:
                                                      
  Test statistic                                 0.000
  Degrees of freedom                                 0

Parameter Estimates:

  Standard errors                             Standard
  Information                                 Expected
  Information saturated (h1) model          Structured

Regressions:
                    Estimate    Std.Err   z-value  P(&gt;|z|)
  sAA_flow_Before ~                                       
    Prmr2_Dpld (c)      15.731     3.274    4.804    0.000
  TP_Before ~                                             
    Prmr2_Dpld (a)     -19.974    16.867   -1.184    0.236
  sAA_flow_Before ~                                       
    TP_Before  (b)       0.053     0.023    2.292    0.022

Variances:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
   .sAA_flow_Befor   9158.083  1570.598    5.831    0.000
   .TP_Before      248016.549 42534.487    5.831    0.000

Defined Parameters:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
    indirect1          -1.067     1.014   -1.052    0.293
    total              14.664     3.364    4.359    0.000</code></pre>
<ul>
<li><strong>Path ‘c’: Prmr2_Dpld –&gt; sAA_flow_Before</strong>
<ul>
<li>Estimate: 15.731</li>
<li>P-value: &lt; 0.001</li>
<li><strong>Interpretation</strong>: There is a significant direct
effect. As “Prmr2_Dpld” increases, “sAA_flow_Before” also significantly
increases.</li>
</ul></li>
<li><strong>Path ‘a’: Prmr2_Dpld –&gt; TP_Before</strong>
<ul>
<li>Estimate: -19.974</li>
<li>P-value: 0.236</li>
<li><strong>Interpretation</strong>: This effect is not statistically
significant. “Prmr2_Dpld” does not significantly influence
“TP_Before”.</li>
</ul></li>
<li><strong>Path ‘b’: TP_Before –&gt; sAA_flow_Before</strong>
<ul>
<li>Estimate: 0.053</li>
<li>P-value: 0.022</li>
<li><strong>Interpretation</strong>: There is a significant direct
effect. As “TP_Before” increases, “sAA_flow_Before” also significantly
increases.</li>
</ul></li>
<li><strong>Indirect effect: Prmr2_Dpld –&gt; TP_Before –&gt;
sAA_flow_Before</strong>
<ul>
<li>Estimate: -1.067</li>
<li>P-value: 0.293</li>
<li><strong>Interpretation</strong>: There is no significant indirect
(mediated) effect. “TP_Before” does not mediate the relationship between
“Prmr2_Dpld” and “sAA_flow_Before”.</li>
</ul></li>
<li><strong>Total effect: Prmr2_Dpld –&gt; sAA_flow_Before</strong>
<ul>
<li>Estimate: 14.664</li>
<li>P-value: &lt; 0.001</li>
<li><strong>Interpretation</strong>: There is a significant overall
effect of “Prmr2_Dpld” on “sAA_flow_Before”.</li>
</ul></li>
<li><strong>Variances</strong>: There’s significant variability in both
“sAA_flow_Before” and “TP_Before” (p &lt; 0.001).</li>
</ul>
<p><strong>Conclusion</strong>: The best model would suggest that
“TP_Before” does not play a mediatory role in the relationship between
“Prmr2_Dpld” and “sAA_flow_Before”.</p>
<pre class="r"><code>layout(matrix(1:2, nrow = 1, byrow = TRUE))
# Tags/labels for each plot
layout_title &lt;- c(&quot;sAA Flow Primer 1 - Before&quot;, &quot;sAA Flow Primer 2 - Before&quot;)

# Create the first plot
plot_sem_model(fit_before_sAA_flow_Primer1)
title(layout_title[1])

# Create the second plot
plot_sem_model(fit_before_sAA_flow_Primer2)
title(layout_title[2])</code></pre>
<p><img src="figure/analysisv2.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/b5db62c16564a512ad4937891c3ee62e7802e59e/docs/figure/analysisv2.Rmd/unnamed-chunk-3-1.png" target="_blank">b5db62c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/6e019773d11bd25f82f5d91be75b101a2da374d8/docs/figure/analysisv2.Rmd/unnamed-chunk-3-1.png" target="_blank">6e01977</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/170c8ef65fdd1fd4c40944e49aab0563198a7516/docs/figure/analysisv2.Rmd/unnamed-chunk-3-1.png" target="_blank">170c8ef</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/07ec9306556753cb5847b51e0594ab20e52e480d/docs/figure/analysisv2.Rmd/unnamed-chunk-3-1.png" target="_blank">07ec930</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="after-1" class="section level4" number="2.1.2.2">
<h4><span class="header-section-number">2.1.2.2</span> After</h4>
<pre class="r"><code># Run the mediation model for Primer1
fit_after_sAA_flow_Primer1 &lt;- run_mediation_model_Primer1_sAA_flow_TP_mediator(df_saliva_lm, &#39;After&#39;)
# Summarize the results
cat(&quot;Summary for Primer1 Model:\n&quot;)</code></pre>
<pre><code>Summary for Primer1 Model:</code></pre>
<pre class="r"><code>summary(fit_after_sAA_flow_Primer1)</code></pre>
<pre><code>lavaan 0.6.15 ended normally after 1 iteration

  Estimator                                         ML
  Optimization method                           NLMINB
  Number of model parameters                         5

  Number of observations                            68

Model Test User Model:
                                                      
  Test statistic                                 0.000
  Degrees of freedom                                 0

Parameter Estimates:

  Standard errors                             Standard
  Information                                 Expected
  Information saturated (h1) model          Structured

Regressions:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
  sAA_flow_After ~                                       
    Prmr1_Dpld (c)      8.724     1.659    5.260    0.000
  TP_After ~                                             
    Prmr1_Dpld (a)     14.561     9.854    1.478    0.139
  sAA_flow_After ~                                       
    TP_After   (b)     -0.033     0.020   -1.633    0.102

Variances:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
   .sAA_flow_After   3505.031   601.108    5.831    0.000
   .TP_After       127687.632 21898.248    5.831    0.000

Defined Parameters:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
    indirect1          -0.478     0.436   -1.096    0.273
    total               8.246     1.664    4.955    0.000</code></pre>
<ul>
<li><strong>Path ‘c’: Prmr1_Dpld –&gt; sAA_flow_After</strong>
<ul>
<li>Estimate: 8.724</li>
<li>P-value: &lt; 0.001</li>
<li><strong>Interpretation</strong>: There’s a significant direct
effect. As “Prmr1_Dpld” increases, “sAA_flow_After” also significantly
increases.</li>
</ul></li>
<li><strong>Path ‘a’: Prmr1_Dpld –&gt; TP_After</strong>
<ul>
<li>Estimate: 14.561</li>
<li>P-value: 0.139</li>
<li><strong>Interpretation</strong>: The effect is not statistically
significant. This means “Prmr1_Dpld” doesn’t significantly influence
“TP_After”.</li>
</ul></li>
<li><strong>Path ‘b’: TP_After –&gt; sAA_flow_After</strong>
<ul>
<li>Estimate: -0.033</li>
<li>P-value: 0.102</li>
<li><strong>Interpretation</strong>: The effect is not statistically
significant. As such, “TP_After” doesn’t significantly influence
“sAA_flow_After”.</li>
</ul></li>
<li><strong>Indirect effect: Prmr1_Dpld –&gt; TP_After –&gt;
sAA_flow_After</strong>
<ul>
<li>Estimate: -0.478</li>
<li>P-value: 0.273</li>
<li><strong>Interpretation</strong>: There is no significant indirect
(mediated) effect. As such, “TP_After” doesn’t mediate the relationship
between “Prmr1_Dpld” and “sAA_flow_After”.</li>
</ul></li>
<li><strong>Total effect: Prmr1_Dpld –&gt; sAA_flow_After</strong>
<ul>
<li>Estimate: 8.246</li>
<li>P-value: &lt; 0.001</li>
<li><strong>Interpretation</strong>: There is a significant overall
effect of “Prmr1_Dpld” on “sAA_flow_After”.</li>
</ul></li>
<li><strong>Variances</strong>: Both “sAA_flow_After” and “TP_After”
show significant variability (p &lt; 0.001).</li>
</ul>
<p><strong>Conclusion</strong>: The best model suggests “TP_After”
doesn’t play a mediatory role in the relationship between “Prmr1_Dpld”
and “sAA_flow_After”.</p>
<pre class="r"><code># Run the mediation model for Primer2
fit_after_sAA_flow_Primer2 &lt;- run_mediation_model_Primer2_sAA_flow_TP_mediator(df_saliva_lm, &#39;After&#39;)
# Summarize the results
cat(&quot;\n\nSummary for Primer2 Model:\n&quot;)</code></pre>
<pre><code>

Summary for Primer2 Model:</code></pre>
<pre class="r"><code>summary(fit_after_sAA_flow_Primer2)</code></pre>
<pre><code>lavaan 0.6.15 ended normally after 10 iterations

  Estimator                                         ML
  Optimization method                           NLMINB
  Number of model parameters                         5

  Number of observations                            68

Model Test User Model:
                                                      
  Test statistic                                 0.000
  Degrees of freedom                                 0

Parameter Estimates:

  Standard errors                             Standard
  Information                                 Expected
  Information saturated (h1) model          Structured

Regressions:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
  sAA_flow_After ~                                       
    Prmr2_Dpld (c)      9.777     2.064    4.736    0.000
  TP_After ~                                             
    Prmr2_Dpld (a)      4.573    12.283    0.372    0.710
  sAA_flow_After ~                                       
    TP_After   (b)     -0.019     0.020   -0.910    0.363

Variances:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
   .sAA_flow_After   3708.097   635.933    5.831    0.000
   .TP_After       131520.028 22555.499    5.831    0.000

Defined Parameters:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
    indirect1          -0.085     0.246   -0.345    0.730
    total               9.692     2.075    4.671    0.000</code></pre>
<p>Here’s the interpretation for the analysis:</p>
<ul>
<li><strong>Path ‘c’: Prmr2_Dpld –&gt; sAA_flow_After</strong>
<ul>
<li>Estimate: 9.777</li>
<li>P-value: &lt; 0.001</li>
<li><strong>Interpretation</strong>: There’s a significant direct
effect. As “Prmr2_Dpld” increases, “sAA_flow_After” also significantly
increases.</li>
</ul></li>
<li><strong>Path ‘a’: Prmr2_Dpld –&gt; TP_After</strong>
<ul>
<li>Estimate: 4.573</li>
<li>P-value: 0.710</li>
<li><strong>Interpretation</strong>: This effect is not statistically
significant, implying “Prmr2_Dpld” does not significantly influence
“TP_After”.</li>
</ul></li>
<li><strong>Path ‘b’: TP_After –&gt; sAA_flow_After</strong>
<ul>
<li>Estimate: -0.019</li>
<li>P-value: 0.363</li>
<li><strong>Interpretation</strong>: The effect is not statistically
significant. As such, “TP_After” doesn’t significantly influence
“sAA_flow_After”.</li>
</ul></li>
<li><strong>Indirect effect: Prmr2_Dpld –&gt; TP_After –&gt;
sAA_flow_After</strong>
<ul>
<li>Estimate: -0.085</li>
<li>P-value: 0.730</li>
<li><strong>Interpretation</strong>: There’s no significant indirect
(mediated) effect. “TP_After” doesn’t mediate the relationship between
“Prmr2_Dpld” and “sAA_flow_After”.</li>
</ul></li>
<li><strong>Total effect: Prmr2_Dpld –&gt; sAA_flow_After</strong>
<ul>
<li>Estimate: 9.692</li>
<li>P-value: &lt; 0.001</li>
<li><strong>Interpretation</strong>: There is a significant overall
effect of “Prmr2_Dpld” on “sAA_flow_After”.</li>
</ul></li>
<li><strong>Variances</strong>: There’s significant variability in both
“sAA_flow_After” and “TP_After” (p &lt; 0.001).</li>
</ul>
<p><strong>Conclusion</strong>: The model indicates “TP_After” doesn’t
play a mediatory role in the relationship between “Prmr2_Dpld” and
“sAA_flow_After”.</p>
</div>
<div id="change-1" class="section level4" number="2.1.2.3">
<h4><span class="header-section-number">2.1.2.3</span> Change</h4>
<pre class="r"><code># Run the mediation model for Primer1
fit_change_sAA_flow_Primer1 &lt;- run_mediation_model_Primer1_sAA_flow_TP_mediator(df_saliva_lm, &#39;Change&#39;)
# Summarize the results
cat(&quot;Summary for Primer1 Model:\n&quot;)</code></pre>
<pre><code>Summary for Primer1 Model:</code></pre>
<pre class="r"><code>summary(fit_change_sAA_flow_Primer1)</code></pre>
<pre><code>lavaan 0.6.15 ended normally after 1 iteration

  Estimator                                         ML
  Optimization method                           NLMINB
  Number of model parameters                         5

  Number of observations                            68

Model Test User Model:
                                                      
  Test statistic                                 0.000
  Degrees of freedom                                 0

Parameter Estimates:

  Standard errors                             Standard
  Information                                 Expected
  Information saturated (h1) model          Structured

Regressions:
                    Estimate    Std.Err   z-value  P(&gt;|z|)
  sAA_flow_Change ~                                       
    Prmr1_Dpld (c)      -5.578     1.545   -3.612    0.000
  TP_Change ~                                             
    Prmr1_Dpld (a)      13.622    13.870    0.982    0.326
  sAA_flow_Change ~                                       
    TP_Change  (b)       0.030     0.013    2.212    0.027

Variances:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
   .sAA_flow_Chang   3093.383   530.511    5.831    0.000
   .TP_Change      253012.389 43391.267    5.831    0.000

Defined Parameters:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
    indirect1           0.404     0.450    0.898    0.369
    total              -5.174     1.588   -3.258    0.001</code></pre>
<ul>
<li><strong>Path ‘c’: Prmr1_Dpld –&gt; sAA_flow_Change</strong>
<ul>
<li>Estimate: -5.578</li>
<li>P-value: &lt; 0.001</li>
<li><strong>Interpretation</strong>: There’s a significant direct
effect. As “Prmr1_Dpld” increases, “sAA_flow_Change” significantly
decreases.</li>
</ul></li>
<li><strong>Path ‘a’: Prmr1_Dpld –&gt; TP_Change</strong>
<ul>
<li>Estimate: 13.622</li>
<li>P-value: 0.326</li>
<li><strong>Interpretation</strong>: This effect is not statistically
significant. This implies “Prmr1_Dpld” does not significantly influence
“TP_Change”.</li>
</ul></li>
<li><strong>Path ‘b’: TP_Change –&gt; sAA_flow_Change</strong>
<ul>
<li>Estimate: 0.030</li>
<li>P-value: 0.027</li>
<li><strong>Interpretation</strong>: There’s a significant direct
effect. As “TP_Change” increases, “sAA_flow_Change” also significantly
increases.</li>
</ul></li>
<li><strong>Indirect effect: Prmr1_Dpld –&gt; TP_Change –&gt;
sAA_flow_Change</strong>
<ul>
<li>Estimate: 0.404</li>
<li>P-value: 0.369</li>
<li><strong>Interpretation</strong>: There’s no significant indirect
(mediated) effect. “TP_Change” doesn’t mediate the relationship between
“Prmr1_Dpld” and “sAA_flow_Change”.</li>
</ul></li>
<li><strong>Total effect: Prmr1_Dpld –&gt; sAA_flow_Change</strong>
<ul>
<li>Estimate: -5.174</li>
<li>P-value: 0.001</li>
<li><strong>Interpretation</strong>: There is a significant overall
effect of “Prmr1_Dpld” on “sAA_flow_Change”.</li>
</ul></li>
<li><strong>Variances</strong>: Both “sAA_flow_Change” and “TP_Change”
show significant variability (p &lt; 0.001).</li>
</ul>
<p><strong>Conclusion</strong>: The best model suggests that “TP_Change”
does not mediate the relationship between “Prmr1_Dpld” and
“sAA_flow_Change”.</p>
<pre class="r"><code># Run the mediation model for Primer2
fit_change_sAA_flow_Primer2 &lt;- run_mediation_model_Primer2_sAA_flow_TP_mediator(df_saliva_lm, &#39;Change&#39;)
# Summarize the results
cat(&quot;\n\nSummary for Primer2 Model:\n&quot;)</code></pre>
<pre><code>

Summary for Primer2 Model:</code></pre>
<pre class="r"><code>summary(fit_change_sAA_flow_Primer2)</code></pre>
<pre><code>lavaan 0.6.15 ended normally after 1 iteration

  Estimator                                         ML
  Optimization method                           NLMINB
  Number of model parameters                         5

  Number of observations                            68

Model Test User Model:
                                                      
  Test statistic                                 0.000
  Degrees of freedom                                 0

Parameter Estimates:

  Standard errors                             Standard
  Information                                 Expected
  Information saturated (h1) model          Structured

Regressions:
                    Estimate    Std.Err   z-value  P(&gt;|z|)
  sAA_flow_Change ~                                       
    Prmr2_Dpld (c)      -5.732     1.969   -2.911    0.004
  TP_Change ~                                             
    Prmr2_Dpld (a)      24.547    16.896    1.453    0.146
  sAA_flow_Change ~                                       
    TP_Change  (b)       0.031     0.014    2.225    0.026

Variances:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
   .sAA_flow_Chang   3278.224   562.211    5.831    0.000
   .TP_Change      248875.953 42681.874    5.831    0.000

Defined Parameters:
                   Estimate    Std.Err   z-value  P(&gt;|z|)
    indirect1           0.760     0.625    1.216    0.224
    total              -4.972     2.009   -2.475    0.013</code></pre>
<ul>
<li><strong>Path ‘c’: Prmr2_Dpld –&gt; sAA_flow_Change</strong>
<ul>
<li>Estimate: -5.732</li>
<li>P-value: 0.004</li>
<li><strong>Interpretation</strong>: There’s a significant direct
effect. As “Prmr2_Dpld” increases, “sAA_flow_Change” significantly
decreases.</li>
</ul></li>
<li><strong>Path ‘a’: Prmr2_Dpld –&gt; TP_Change</strong>
<ul>
<li>Estimate: 24.547</li>
<li>P-value: 0.146</li>
<li><strong>Interpretation</strong>: The effect is not statistically
significant. This means “Prmr2_Dpld” does not significantly influence
“TP_Change”.</li>
</ul></li>
<li><strong>Path ‘b’: TP_Change –&gt; sAA_flow_Change</strong>
<ul>
<li>Estimate: 0.031</li>
<li>P-value: 0.026</li>
<li><strong>Interpretation</strong>: There’s a significant direct
effect. As “TP_Change” increases, “sAA_flow_Change” also significantly
increases.</li>
</ul></li>
<li><strong>Indirect effect: Prmr2_Dpld –&gt; TP_Change –&gt;
sAA_flow_Change</strong>
<ul>
<li>Estimate: 0.760</li>
<li>P-value: 0.224</li>
<li><strong>Interpretation</strong>: There is no significant indirect
(mediated) effect, meaning “TP_Change” doesn’t mediate the relationship
between “Prmr2_Dpld” and “sAA_flow_Change”.</li>
</ul></li>
<li><strong>Total effect: Prmr2_Dpld –&gt; sAA_flow_Change</strong>
<ul>
<li>Estimate: -4.972</li>
<li>P-value: 0.013</li>
<li><strong>Interpretation</strong>: There is a significant overall
effect of “Prmr2_Dpld” on “sAA_flow_Change”.</li>
</ul></li>
<li><strong>Variances</strong>: Both “sAA_flow_Change” and “TP_Change”
show significant variability (p &lt; 0.001).</li>
</ul>
<p><strong>Conclusion</strong>: The best model suggests that “TP_Change”
does not mediate the relationship between “Prmr2_Dpld” and
“sAA_flow_Change”.</p>
<pre class="r"><code>layout(matrix(1:2, nrow = 1, byrow = TRUE))
# Tags/labels for each plot
layout_title &lt;- c(&quot;sAA Flow Primer 1 - Change&quot;, &quot;sAA Flow Primer 2 - Change&quot;)

# Create the first plot
plot_sem_model(fit_change_sAA_flow_Primer1)
title(layout_title[1])

# Create the second plot
plot_sem_model(fit_change_sAA_flow_Primer2)
title(layout_title[2])</code></pre>
<p><img src="figure/analysisv2.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/b5db62c16564a512ad4937891c3ee62e7802e59e/docs/figure/analysisv2.Rmd/unnamed-chunk-8-1.png" target="_blank">b5db62c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/eb2e32f709faaf9e78b11c8e22c1e29ade2d49bc/docs/figure/analysisv2.Rmd/unnamed-chunk-8-1.png" target="_blank">eb2e32f</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/170c8ef65fdd1fd4c40944e49aab0563198a7516/docs/figure/analysisv2.Rmd/unnamed-chunk-8-1.png" target="_blank">170c8ef</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/07ec9306556753cb5847b51e0594ab20e52e480d/docs/figure/analysisv2.Rmd/unnamed-chunk-8-1.png" target="_blank">07ec930</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="relationship-between-cnv-saa-and-citrus-perception"
class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Relationship between
CNV, sAA, and citrus perception</h2>
<pre class="r"><code># Generalized function for mediation model including Flow_Rate_mean
run_citrus_model &lt;- function(df, timepoint_or_change) {
  mediation_model &lt;- paste(&quot;
    # direct effect
    Citrus_Sum_120sec ~ c*Primer2_Diploid
    # mediators
    sAA_&quot;, timepoint_or_change, &quot; ~ a*Primer2_Diploid
    Flow_Rate_mean ~ m*Primer2_Diploid
    Citrus_Sum_120sec ~ b*sAA_&quot;, timepoint_or_change, &quot; + n*Flow_Rate_mean
    # indirect effects (a*b and m*n)
    indirect1 := a*b
    indirect2 := m*n
    # total effect
    total := c + (a*b) + (m*n)
  &quot;, sep=&quot;&quot;)
  

  # Fit the model
  fit &lt;- sem(mediation_model, data=df)

  # Return the model
  return(fit)
}</code></pre>
<div id="before-2" class="section level3" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Before</h3>
<pre class="r"><code>citrus_model_before &lt;- run_citrus_model(df_saliva_lm, &#39;Before&#39;)

summary(citrus_model_before)</code></pre>
<pre><code>lavaan 0.6.15 ended normally after 1 iteration

  Estimator                                         ML
  Optimization method                           NLMINB
  Number of model parameters                         8

  Number of observations                            68

Model Test User Model:
                                                      
  Test statistic                                 0.765
  Degrees of freedom                                 1
  P-value (Chi-square)                           0.382

Parameter Estimates:

  Standard errors                             Standard
  Information                                 Expected
  Information saturated (h1) model          Structured

Regressions:
                      Estimate  Std.Err  z-value  P(&gt;|z|)
  Citrus_Sum_120sec ~                                    
    Prmr2_Dpld (c)       1.397    1.622    0.861    0.389
  sAA_Before ~                                           
    Prmr2_Dpld (a)      11.121    2.940    3.783    0.000
  Flow_Rate_mean ~                                       
    Prmr2_Dpld (m)       0.027    0.013    2.040    0.041
  Citrus_Sum_120sec ~                                    
    sAA_Before (b)      -0.021    0.059   -0.348    0.728
    Flow_Rt_mn (n)     -18.025   13.118   -1.374    0.169

Variances:
                   Estimate  Std.Err  z-value  P(&gt;|z|)
   .Citrs_Sm_120sc 1802.768  309.172    5.831    0.000
   .sAA_Before     7532.847 1291.873    5.831    0.000
   .Flow_Rate_mean    0.154    0.026    5.831    0.000

Defined Parameters:
                   Estimate  Std.Err  z-value  P(&gt;|z|)
    indirect1        -0.230    0.663   -0.347    0.729
    indirect2        -0.489    0.429   -1.140    0.254
    total             0.678    1.459    0.465    0.642</code></pre>
<pre class="r"><code>plot_sem_model(citrus_model_before)</code></pre>
<p><img src="figure/analysisv2.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/b5db62c16564a512ad4937891c3ee62e7802e59e/docs/figure/analysisv2.Rmd/unnamed-chunk-11-1.png" target="_blank">b5db62c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/6e019773d11bd25f82f5d91be75b101a2da374d8/docs/figure/analysisv2.Rmd/unnamed-chunk-11-1.png" target="_blank">6e01977</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li><strong>Path ‘c’: Prmr2_Dpld –&gt; Citrus_Sum_120sec</strong>
<ul>
<li>Estimate: 1.397</li>
<li>P-value: 0.389</li>
<li><strong>Interpretation</strong>: The effect is not statistically
significant, suggesting that the increase in “Prmr2_Dpld” does not
significantly impact “Citrus_Sum_120sec”.</li>
</ul></li>
<li><strong>Path ‘a’: Prmr2_Dpld –&gt; sAA_Before</strong>
<ul>
<li>Estimate: 11.121</li>
<li>P-value: &lt;0.001</li>
<li><strong>Interpretation</strong>: There’s a statistically significant
effect, implying that as “Prmr2_Dpld” increases, “sAA_Before” also
significantly increases.</li>
</ul></li>
<li><strong>Path ‘m’: Prmr2_Dpld –&gt; Flow_Rate_mean</strong>
<ul>
<li>Estimate: 0.027</li>
<li>P-value: 0.041</li>
<li><strong>Interpretation</strong>: There’s a significant effect
suggesting that an increase in “Prmr2_Dpld” leads to a significant
increase in “Flow_Rate_mean”.</li>
</ul></li>
<li><strong>Path ‘b’: sAA_Before –&gt; Citrus_Sum_120sec</strong>
<ul>
<li>Estimate: -0.021</li>
<li>P-value: 0.728</li>
<li><strong>Interpretation</strong>: This effect isn’t statistically
significant. “sAA_Before” does not significantly influence
“Citrus_Sum_120sec”.</li>
</ul></li>
<li><strong>Path ‘n’: Flow_Rt_mn –&gt; Citrus_Sum_120sec</strong>
<ul>
<li>Estimate: -18.025</li>
<li>P-value: 0.169</li>
<li><strong>Interpretation</strong>: The effect is not statistically
significant, suggesting “Flow_Rt_mn” doesn’t significantly influence
“Citrus_Sum_120sec”.</li>
</ul></li>
<li><strong>Indirect Paths:</strong>
<ul>
<li><strong>First indirect effect: Prmr2_Dpld –&gt; sAA_Before –&gt;
Citrus_Sum_120sec</strong>
<ul>
<li>Estimate: -0.230</li>
<li>P-value: 0.729</li>
<li><strong>Interpretation</strong>: There is no significant indirect
effect, meaning “sAA_Before” doesn’t mediate the relationship between
“Prmr2_Dpld” and “Citrus_Sum_120sec”.</li>
</ul></li>
<li><strong>Second indirect effect: Prmr2_Dpld –&gt; Flow_Rate_mean
–&gt; Citrus_Sum_120sec</strong>
<ul>
<li>Estimate: -0.489</li>
<li>P-value: 0.254</li>
<li><strong>Interpretation</strong>: No significant indirect effect is
found here either, suggesting “Flow_Rate_mean” doesn’t mediate the
relationship between “Prmr2_Dpld” and “Citrus_Sum_120sec”.</li>
</ul></li>
</ul></li>
<li><strong>Total effect: Prmr2_Dpld –&gt; Citrus_Sum_120sec</strong>
<ul>
<li>Estimate: 0.678</li>
<li>P-value: 0.642</li>
<li><strong>Interpretation</strong>: The total effect isn’t
statistically significant, hinting “Prmr2_Dpld” does not significantly
influence “Citrus_Sum_120sec”.</li>
</ul></li>
<li><strong>Variances</strong>: The variance of each of the variables
“Citrs_Sm_120sc”, “sAA_Before”, and “Flow_Rate_mean” is significant
suggesting there’s considerable variation in these variables.</li>
</ul>
<p><strong>Conclusion</strong>: Both “sAA_Before” and “Flow_Rate_mean”
do not appear to play significant mediating roles in the relationship
between “Prmr2_Dpld” and “Citrus_Sum_120sec”.</p>
</div>
<div id="after-2" class="section level3" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> After</h3>
<pre class="r"><code>citrus_model_after &lt;- run_citrus_model(df_saliva_lm, &#39;After&#39;)

summary(citrus_model_after)</code></pre>
<pre><code>lavaan 0.6.15 ended normally after 2 iterations

  Estimator                                         ML
  Optimization method                           NLMINB
  Number of model parameters                         8

  Number of observations                            68

Model Test User Model:
                                                      
  Test statistic                                 0.950
  Degrees of freedom                                 1
  P-value (Chi-square)                           0.330

Parameter Estimates:

  Standard errors                             Standard
  Information                                 Expected
  Information saturated (h1) model          Structured

Regressions:
                      Estimate  Std.Err  z-value  P(&gt;|z|)
  Citrus_Sum_120sec ~                                    
    Prmr2_Dpld (c)       0.916    1.647    0.556    0.578
  sAA_After ~                                            
    Prmr2_Dpld (a)       6.438    1.561    4.125    0.000
  Flow_Rate_mean ~                                       
    Prmr2_Dpld (m)       0.027    0.013    2.040    0.041
  Citrus_Sum_120sec ~                                    
    sAA_After  (b)       0.035    0.112    0.312    0.755
    Flow_Rt_mn (n)     -17.060   13.121   -1.300    0.194

Variances:
                   Estimate  Std.Err  z-value  P(&gt;|z|)
   .Citrs_Sm_120sc 1803.394  309.279    5.831    0.000
   .sAA_After      2123.637  364.201    5.831    0.000
   .Flow_Rate_mean    0.154    0.026    5.831    0.000

Defined Parameters:
                   Estimate  Std.Err  z-value  P(&gt;|z|)
    indirect1         0.225    0.722    0.311    0.755
    indirect2        -0.463    0.422   -1.097    0.273
    total             0.678    1.457    0.466    0.642</code></pre>
<pre class="r"><code>plot_sem_model(citrus_model_after)</code></pre>
<p><img src="figure/analysisv2.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/b5db62c16564a512ad4937891c3ee62e7802e59e/docs/figure/analysisv2.Rmd/unnamed-chunk-13-1.png" target="_blank">b5db62c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/6e019773d11bd25f82f5d91be75b101a2da374d8/docs/figure/analysisv2.Rmd/unnamed-chunk-13-1.png" target="_blank">6e01977</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/170c8ef65fdd1fd4c40944e49aab0563198a7516/docs/figure/analysisv2.Rmd/unnamed-chunk-13-1.png" target="_blank">170c8ef</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/07ec9306556753cb5847b51e0594ab20e52e480d/docs/figure/analysisv2.Rmd/unnamed-chunk-13-1.png" target="_blank">07ec930</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li><strong>Path ‘c’: Prmr2_Dpld –&gt; Citrus_Sum_120sec</strong>
<ul>
<li>Estimate: 0.916</li>
<li>P-value: 0.578</li>
<li><strong>Interpretation</strong>: The effect is not statistically
significant, suggesting “Prmr2_Dpld” does not significantly impact
“Citrus_Sum_120sec”.</li>
</ul></li>
<li><strong>Path ‘a’: Prmr2_Dpld –&gt; sAA_After</strong>
<ul>
<li>Estimate: 6.438</li>
<li>P-value: &lt;0.001</li>
<li><strong>Interpretation</strong>: There’s a statistically significant
effect, suggesting that as “Prmr2_Dpld” increases, “sAA_After” also
significantly increases.</li>
</ul></li>
<li><strong>Path ‘m’: Prmr2_Dpld –&gt; Flow_Rate_mean</strong>
<ul>
<li>Estimate: 0.027</li>
<li>P-value: 0.041</li>
<li><strong>Interpretation</strong>: There’s a significant effect,
suggesting an increase in “Prmr2_Dpld” leads to a significant increase
in “Flow_Rate_mean”.</li>
</ul></li>
<li><strong>Path ‘b’: sAA_After –&gt; Citrus_Sum_120sec</strong>
<ul>
<li>Estimate: 0.035</li>
<li>P-value: 0.755</li>
<li><strong>Interpretation</strong>: This effect isn’t statistically
significant, implying “sAA_After” doesn’t significantly influence
“Citrus_Sum_120sec”.</li>
</ul></li>
<li><strong>Path ‘n’: Flow_Rt_mn –&gt; Citrus_Sum_120sec</strong>
<ul>
<li>Estimate: -17.060</li>
<li>P-value: 0.194</li>
<li><strong>Interpretation</strong>: The effect is not statistically
significant, suggesting “Flow_Rt_mn” does not significantly influence
“Citrus_Sum_120sec”.</li>
</ul></li>
<li><strong>Indirect Paths:</strong>
<ul>
<li><strong>First indirect effect: Prmr2_Dpld –&gt; sAA_After –&gt;
Citrus_Sum_120sec</strong>
<ul>
<li>Estimate: 0.225</li>
<li>P-value: 0.755</li>
<li><strong>Interpretation</strong>: There is no significant indirect
effect, implying “sAA_After” doesn’t mediate the relationship between
“Prmr2_Dpld” and “Citrus_Sum_120sec”.</li>
</ul></li>
<li><strong>Second indirect effect: Prmr2_Dpld –&gt; Flow_Rate_mean
–&gt; Citrus_Sum_120sec</strong>
<ul>
<li>Estimate: -0.463</li>
<li>P-value: 0.273</li>
<li><strong>Interpretation</strong>: No significant indirect effect is
found here either, suggesting “Flow_Rate_mean” doesn’t mediate the
relationship between “Prmr2_Dpld” and “Citrus_Sum_120sec”.</li>
</ul></li>
</ul></li>
<li><strong>Total effect: Prmr2_Dpld –&gt; Citrus_Sum_120sec</strong>
<ul>
<li>Estimate: 0.678</li>
<li>P-value: 0.642</li>
<li><strong>Interpretation</strong>: The total effect is not
statistically significant, suggesting “Prmr2_Dpld” does not
significantly influence “Citrus_Sum_120sec”.</li>
</ul></li>
<li><strong>Variances</strong>: Both “Citrs_Sm_120sc”, “sAA_After”, and
“Flow_Rate_mean” show significant variability (p &lt; 0.001).</li>
</ul>
<p><strong>Conclusion</strong>: Both “sAA_After” and “Flow_Rate_mean” do
not act as significant mediators in the relationship between
“Prmr2_Dpld” and “Citrus_Sum_120sec”.</p>
</div>
<div id="change-2" class="section level3" number="2.2.3">
<h3><span class="header-section-number">2.2.3</span> Change</h3>
<pre class="r"><code>citrus_model_change &lt;- run_citrus_model(df_saliva_lm, &#39;Change&#39;)

summary(citrus_model_change)</code></pre>
<pre><code>lavaan 0.6.15 ended normally after 1 iteration

  Estimator                                         ML
  Optimization method                           NLMINB
  Number of model parameters                         8

  Number of observations                            68

Model Test User Model:
                                                      
  Test statistic                                 0.330
  Degrees of freedom                                 1
  P-value (Chi-square)                           0.566

Parameter Estimates:

  Standard errors                             Standard
  Information                                 Expected
  Information saturated (h1) model          Structured

Regressions:
                      Estimate  Std.Err  z-value  P(&gt;|z|)
  Citrus_Sum_120sec ~                                    
    Prmr2_Dpld (c)       1.541    1.541    1.000    0.317
  sAA_Change ~                                           
    Prmr2_Dpld (a)      -4.683    1.828   -2.561    0.010
  Flow_Rate_mean ~                                       
    Prmr2_Dpld (m)       0.027    0.013    2.040    0.041
  Citrus_Sum_120sec ~                                    
    sAA_Change (b)       0.078    0.095    0.824    0.410
    Flow_Rt_mn (n)     -18.291   13.065   -1.400    0.162

Variances:
                   Estimate  Std.Err  z-value  P(&gt;|z|)
   .Citrs_Sm_120sc 1788.184  306.671    5.831    0.000
   .sAA_Change     2913.616  499.681    5.831    0.000
   .Flow_Rate_mean    0.154    0.026    5.831    0.000

Defined Parameters:
                   Estimate  Std.Err  z-value  P(&gt;|z|)
    indirect1        -0.367    0.467   -0.784    0.433
    indirect2        -0.496    0.430   -1.154    0.248
    total             0.678    1.460    0.465    0.642</code></pre>
<pre class="r"><code>plot_sem_model(citrus_model_change)</code></pre>
<p><img src="figure/analysisv2.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/b5db62c16564a512ad4937891c3ee62e7802e59e/docs/figure/analysisv2.Rmd/unnamed-chunk-15-1.png" target="_blank">b5db62c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/170c8ef65fdd1fd4c40944e49aab0563198a7516/docs/figure/analysisv2.Rmd/unnamed-chunk-15-1.png" target="_blank">170c8ef</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/07ec9306556753cb5847b51e0594ab20e52e480d/docs/figure/analysisv2.Rmd/unnamed-chunk-15-1.png" target="_blank">07ec930</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The interpretations of your results are as follows:</p>
<ul>
<li><strong>Path ‘c’: Prmr2_Dpld –&gt; Citrus_Sum_120sec</strong>
<ul>
<li>Estimate: 1.541</li>
<li>P-value: 0.317</li>
<li><strong>Interpretation</strong>: The effect is not statistically
significant, suggesting that “Prmr2_Dpld” does not significantly impact
“Citrus_Sum_120sec”.</li>
</ul></li>
<li><strong>Path ‘a’: Prmr2_Dpld –&gt; sAA_Change</strong>
<ul>
<li>Estimate: -4.683</li>
<li>P-value: 0.010</li>
<li><strong>Interpretation</strong>: There’s a statistically significant
effect, suggesting that as “Prmr2_Dpld” increases, “sAA_Change”
significantly decreases.</li>
</ul></li>
<li><strong>Path ‘m’: Prmr2_Dpld –&gt; Flow_Rate_mean</strong>
<ul>
<li>Estimate: 0.027</li>
<li>P-value: 0.041</li>
<li><strong>Interpretation</strong>: There’s a significant effect,
suggesting an increase in “Prmr2_Dpld” leads to a significant increase
in “Flow_Rate_mean”.</li>
</ul></li>
<li><strong>Path ‘b’: sAA_Change –&gt; Citrus_Sum_120sec</strong>
<ul>
<li>Estimate: 0.078</li>
<li>P-value: 0.410</li>
<li><strong>Interpretation</strong>: The effect is not statistically
significant, implying “sAA_Change” doesn’t significantly influence
“Citrus_Sum_120sec”.</li>
</ul></li>
<li><strong>Path ‘n’: Flow_Rt_mn –&gt; Citrus_Sum_120sec</strong>
<ul>
<li>Estimate: -18.291</li>
<li>P-value: 0.162</li>
<li><strong>Interpretation</strong>: The effect is not statistically
significant, suggesting “Flow_Rt_mn” does not significantly influence
“Citrus_Sum_120sec”.</li>
</ul></li>
<li><strong>Indirect Paths:</strong>
<ul>
<li><strong>First indirect effect: Prmr2_Dpld –&gt; sAA_Change –&gt;
Citrus_Sum_120sec</strong>
<ul>
<li>Estimate: -0.367</li>
<li>P-value: 0.433</li>
<li><strong>Interpretation</strong>: There is no significant indirect
effect, suggesting “sAA_Change” doesn’t mediate the relationship between
“Prmr2_Dpld” and “Citrus_Sum_120sec”.</li>
</ul></li>
<li><strong>Second indirect effect: Prmr2_Dpld –&gt; Flow_Rate_mean
–&gt; Citrus_Sum_120sec</strong>
<ul>
<li>Estimate: -0.496</li>
<li>P-value: 0.248</li>
<li><strong>Interpretation</strong>: No significant indirect effect is
found, implying “Flow_Rate_mean” doesn’t mediate the relationship
between “Prmr2_Dpld” and “Citrus_Sum_120sec”.</li>
</ul></li>
</ul></li>
<li><strong>Total effect: Prmr2_Dpld –&gt; Citrus_Sum_120sec</strong>
<ul>
<li>Estimate: 0.678</li>
<li>P-value: 0.642</li>
<li><strong>Interpretation</strong>: The total effect is not
statistically significant, suggesting “Prmr2_Dpld” does not
significantly influence “Citrus_Sum_120sec”.</li>
</ul></li>
<li><strong>Variances</strong>: Both “Citrs_Sm_120sc”, “sAA_Change”, and
“Flow_Rate_mean” show significant variability (p &lt; 0.001).</li>
</ul>
<p><strong>Conclusion</strong>: Both “sAA_Change” and “Flow_Rate_mean”
do not act as significant mediators in the relationship between
“Prmr2_Dpld” and “Citrus_Sum_120sec”.</p>
</div>
</div>
</div>
<div id="flavor-perception" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Flavor perception</h1>
<pre class="r"><code>df_flavor &lt;- read_csv(&quot;data/df_flavor.csv&quot;)</code></pre>
<pre class="r"><code>data_long &lt;- df_flavor %&gt;%
  select(-c(Blinding_Code, Sample_Position)) %&gt;%
  pivot_longer(cols = starts_with(&quot;time_&quot;),
               names_to = &quot;Time&quot;,
               names_prefix = &quot;time_&quot;, 
               values_to = &quot;Value&quot;) %&gt;%
  mutate(Time = as.numeric(str_remove(Time, &quot;s&quot;)),
         Sample_ID = as.factor(Sample_ID),
         Sample_Name = ifelse(Sample_Name == &quot;Starch-Limonene Inclusion Complex&quot;,
                                &quot;Starch-Limonene IC&quot;,
                                Sample_Name),
         Sample_Name = as.factor(Sample_Name),
         Attribute = as.factor(Attribute))</code></pre>
<pre class="r"><code>data_summary &lt;- data_long %&gt;%
  group_by(Sample_Name, Attribute) %&gt;%
  summarise(mean = mean(Value, na.rm = TRUE),
            sd = sd(Value, na.rm = TRUE),
            .groups = &quot;drop&quot;)  # drop grouping afterwards</code></pre>
<pre class="r"><code>data_summary &lt;- data_long %&gt;%
  group_by(Sample_Name, Attribute, Time) %&gt;%
  summarise(mean_value = mean(Value, na.rm = TRUE), .groups = &quot;drop&quot;)


ggplot(data_summary, aes(x = Time, y = mean_value, color = Attribute)) +
  geom_line() +
  facet_wrap(~ Sample_Name) +
  labs(x = &#39;Time&#39;, y = &#39;Proportion Perceiving Flavor&#39;, color = &#39;Attribute&#39;) +
  theme_classic()</code></pre>
<p><img src="figure/analysisv2.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/b5db62c16564a512ad4937891c3ee62e7802e59e/docs/figure/analysisv2.Rmd/unnamed-chunk-19-1.png" target="_blank">b5db62c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/6e019773d11bd25f82f5d91be75b101a2da374d8/docs/figure/analysisv2.Rmd/unnamed-chunk-19-1.png" target="_blank">6e01977</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Pivot data to calculate the difference
data_diff &lt;- data_summary %&gt;%
  pivot_wider(names_from = Sample_Name, values_from = mean_value) %&gt;%
  mutate(Difference = `Free Limonene` - `Starch-Limonene IC`)

# Plot 
ggplot(data_diff, aes(x = Time, y = Difference, color = Attribute)) +
  geom_line() +
  labs(x = &#39;Time&#39;, y = &#39;Difference in Perception&#39;, color = &#39;Attribute&#39;) +
  theme_classic()</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/analysisv2.Rmd/fig-diff-starch-free-1.png" alt="Positive values indicate higher perception in Free Limonene and negative values indicate higher perception in Starch-Limonene IC" width="672" />
<p class="caption">
Positive values indicate higher perception in Free Limonene and negative
values indicate higher perception in Starch-Limonene IC
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig-diff-starch-free-1">
Past versions of fig-diff-starch-free-1.png
</button>
</p>
<div id="fig-fig-diff-starch-free-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/6e019773d11bd25f82f5d91be75b101a2da374d8/docs/figure/analysisv2.Rmd/fig-diff-starch-free-1.png" target="_blank">6e01977</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Joining the data
combined_data &lt;- data_long %&gt;%
  inner_join(df_saliva, by = &quot;Sample_ID&quot;) %&gt;%
  na.omit()</code></pre>
<pre class="r"><code># Categorize gene copy number into two groups: low and high
combined_data_CNV &lt;- combined_data %&gt;% 
  mutate(CNV_group = cut(Primer2_Diploid, breaks = 2, labels = c(&quot;Low&quot;, &quot;High&quot;)))

# Overall proportion perceiving flavor at each time point, by attribute, sample name
flavor_prop_all &lt;- combined_data_CNV %&gt;% 
  group_by(Time, Attribute, Sample_Name) %&gt;%
  summarise(overall_prop = mean(Value), .groups = &quot;drop&quot;)

# Proportions perceiving flavor within each CNV group
flavor_prop_CNV &lt;- combined_data_CNV %&gt;% 
  filter(Value == 1) %&gt;%
  group_by(Time, Attribute, Sample_Name, CNV_group) %&gt;%
  summarise(count_flavor = n(), .groups = &quot;drop&quot;) %&gt;%
  group_by(Time, Attribute, Sample_Name) %&gt;%
  mutate(proportion_flavor = count_flavor / sum(count_flavor))

# Merge both data frames
final_data &lt;- merge(flavor_prop_all, flavor_prop_CNV, by = c(&quot;Time&quot;, &quot;Attribute&quot;, &quot;Sample_Name&quot;))

# plot
ggplot(final_data, aes(x = Time, y = overall_prop, fill = CNV_group)) +
  geom_col(aes(fill = CNV_group, y = overall_prop * proportion_flavor), position = &quot;stack&quot;) +
  facet_grid(Attribute ~ Sample_Name) +
  labs(x = &#39;Time&#39;, y = &#39;Proportion Perceiving Flavor&#39;, fill = &#39;Copy Number Group&#39;)</code></pre>
<p><img src="figure/analysisv2.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/b5db62c16564a512ad4937891c3ee62e7802e59e/docs/figure/analysisv2.Rmd/unnamed-chunk-21-1.png" target="_blank">b5db62c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/6e019773d11bd25f82f5d91be75b101a2da374d8/docs/figure/analysisv2.Rmd/unnamed-chunk-21-1.png" target="_blank">6e01977</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/170c8ef65fdd1fd4c40944e49aab0563198a7516/docs/figure/analysisv2.Rmd/unnamed-chunk-21-1.png" target="_blank">170c8ef</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/07ec9306556753cb5847b51e0594ab20e52e480d/docs/figure/analysisv2.Rmd/unnamed-chunk-21-1.png" target="_blank">07ec930</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Reshape data to wide format
df_wide1 &lt;- data_long %&gt;%
    pivot_wider(
        names_from = c(Sample_Name, Attribute), 
        values_from = Value,
        names_glue = &quot;{Sample_Name}_{Attribute}&quot;
    )</code></pre>
<pre class="r"><code>samples_to_compare &lt;- c(&quot;Free Limonene&quot;, &quot;Starch-Limonene IC&quot;) 

# Generate difference columns
df_wide_diff &lt;- df_wide1
for (attrib in unique(data_long$Attribute)) {
    df_wide_diff &lt;- df_wide_diff %&gt;%
        mutate(
            !!paste0(&quot;Difference_&quot;, attrib) :=
                df_wide_diff[[paste0(samples_to_compare[1], &quot;_&quot;, attrib)]] - 
                df_wide_diff[[paste0(samples_to_compare[2], &quot;_&quot;, attrib)]]
        )
}

df_wide_diff</code></pre>
<pre><code># A tibble: 12,100 × 20
   Sample_ID  Time `Free Limonene_Sweet` `Starch-Limonene IC_Sweet`
   &lt;fct&gt;     &lt;dbl&gt;                 &lt;dbl&gt;                      &lt;dbl&gt;
 1 1             0                     0                          0
 2 1             1                     0                          0
 3 1             2                     0                          0
 4 1             3                     0                          0
 5 1             4                     0                          0
 6 1             5                     0                          0
 7 1             6                     0                          1
 8 1             7                     0                          1
 9 1             8                     0                          1
10 1             9                     0                          1
# ℹ 12,090 more rows
# ℹ 16 more variables: `Free Limonene_Bitter` &lt;dbl&gt;,
#   `Starch-Limonene IC_Bitter` &lt;dbl&gt;, `Free Limonene_Sour` &lt;dbl&gt;,
#   `Starch-Limonene IC_Sour` &lt;dbl&gt;, `Free Limonene_Citrus` &lt;dbl&gt;,
#   `Starch-Limonene IC_Citrus` &lt;dbl&gt;, `Free Limonene_Stale` &lt;dbl&gt;,
#   `Starch-Limonene IC_Stale` &lt;dbl&gt;, `Free Limonene_Cohesive` &lt;dbl&gt;,
#   `Starch-Limonene IC_Cohesive` &lt;dbl&gt;, Difference_Sweet &lt;dbl&gt;, …</code></pre>
<pre class="r"><code>df_saliva_filter &lt;- df_saliva %&gt;% 
  mutate(Flow_Rate_mean = (Flow_Rate_1_g_min + Flow_Rate_2_g_min)/2) %&gt;% 
  select(c(Sample_ID, Primer1_Diploid, Primer2_Diploid, sAA_Before, sAA_After, sAA_Change, TP_Before, TP_After, TP_Change, Flow_Rate_mean, Citrus_Sum_120sec)) %&gt;% 
  na.omit()

combined_data_diff &lt;- df_wide_diff %&gt;%
  inner_join(df_saliva_filter, by = &quot;Sample_ID&quot;) %&gt;% 
  na.omit(Primer2_Diploid) %&gt;% 
  mutate(CNV_group = if_else(Primer2_Diploid &gt; quantile(Primer2_Diploid, probs = .75), &quot;High&quot;, &quot;Low&quot;),
         sAA_After_group = if_else(sAA_After &gt; quantile(sAA_After, probs = .75), &quot;High&quot;, &quot;Low&quot;),
         sAA_Change_group = if_else(sAA_Change &gt; quantile(sAA_Change, probs = .75), &quot;High&quot;, &quot;Low&quot;),
         sAA_flow_Afterrate = sAA_After*Flow_Rate_mean,
         sAA_flow_After_group = if_else(sAA_flow_Afterrate &gt; quantile(sAA_flow_Afterrate, probs = .75), &quot;High&quot;, &quot;Low&quot;))</code></pre>
<pre class="r"><code>plot_perception &lt;- function(data, group_col, group_labels, color_vector = c(&quot;blue&quot;, &quot;red&quot;, &quot;green&quot;)) {
  
  # Ensure group column is a factor
  data[[group_col]] &lt;- as.factor(data[[group_col]])

  # Get the name of the group_col variable for use in labels
  group_col_name &lt;- deparse(substitute(group_col))
  
  # Calculate proportions
  data_plot &lt;- data %&gt;%
    gather(key = &quot;Attribute&quot;, value = &quot;Difference&quot;, starts_with(&quot;Difference_&quot;)) %&gt;% 
    mutate(Positive_Change = Difference &gt; 0,
           Negative_Change = Difference &lt; 0,
           No_Change = Difference == 0) %&gt;%
    group_by(Attribute, !!sym(group_col), Time) %&gt;%
    summarize(Prop_Positive_Change = mean(Positive_Change),
              Prop_Negative_Change = mean(Negative_Change),
              Prop_No_Change = mean(No_Change), .groups = &#39;drop&#39;) %&gt;%
    pivot_longer(starts_with(&quot;Prop_&quot;), names_to = &quot;Change&quot;, values_to = &quot;Proportion&quot;) 

  # Plot
  ggplot(data_plot, aes(x = Time, y = Proportion, color = !!sym(group_col), linetype = Change)) +
    geom_line() +
    scale_y_continuous(labels = scales::percent) +
    scale_color_manual(values = color_vector, labels = group_labels) +
    scale_linetype_manual(values = c(&quot;Prop_Positive_Change&quot; = &quot;solid&quot;, &quot;Prop_Negative_Change&quot; = &quot;dashed&quot;, &quot;Prop_No_Change&quot; = &quot;dotted&quot;)) +
    labs(y = &quot;Proportion of Change in Perception&quot;,
         x = &quot;Time&quot;,
         color = group_col_name, 
         linetype = &quot;Change Type&quot;) +
    theme_minimal() +
    facet_grid(Attribute ~ Change) + 
    ggtitle(&quot;Positive, Negative, and No Change in Flavor Perception Over Time&quot;)
  
}</code></pre>
<pre class="r"><code># Use function
plot_perception(combined_data_diff, &quot;CNV_group&quot;, c(&quot;Low&quot;, &quot;High&quot;))</code></pre>
<p><img src="figure/analysisv2.Rmd/unnamed-chunk-26-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-1">
Past versions of unnamed-chunk-26-1.png
</button>
</p>
<div id="fig-unnamed-chunk-26-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/b5db62c16564a512ad4937891c3ee62e7802e59e/docs/figure/analysisv2.Rmd/unnamed-chunk-26-1.png" target="_blank">b5db62c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/eb2e32f709faaf9e78b11c8e22c1e29ade2d49bc/docs/figure/analysisv2.Rmd/unnamed-chunk-26-1.png" target="_blank">eb2e32f</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Use function
plot_perception(combined_data_diff, &quot;sAA_After_group&quot;, c(&quot;Low&quot;, &quot;High&quot;))</code></pre>
<p><img src="figure/analysisv2.Rmd/unnamed-chunk-27-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-27-1">
Past versions of unnamed-chunk-27-1.png
</button>
</p>
<div id="fig-unnamed-chunk-27-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/b5db62c16564a512ad4937891c3ee62e7802e59e/docs/figure/analysisv2.Rmd/unnamed-chunk-27-1.png" target="_blank">b5db62c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Use function
plot_perception(combined_data_diff, &quot;sAA_Change_group&quot;, c(&quot;Low&quot;, &quot;High&quot;))</code></pre>
<p><img src="figure/analysisv2.Rmd/unnamed-chunk-28-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/b5db62c16564a512ad4937891c3ee62e7802e59e/docs/figure/analysisv2.Rmd/unnamed-chunk-28-1.png" target="_blank">b5db62c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Use function
plot_perception(combined_data_diff, &quot;sAA_flow_After_group&quot;, c(&quot;Low&quot;, &quot;High&quot;))</code></pre>
<p><img src="figure/analysisv2.Rmd/unnamed-chunk-29-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/Goza_AMY_CNV/blob/b5db62c16564a512ad4937891c3ee62e7802e59e/docs/figure/analysisv2.Rmd/unnamed-chunk-29-1.png" target="_blank">b5db62c</a>
</td>
<td>
Tina Lasisi
</td>
<td>
2023-07-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.2 (2022-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.4

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] semPlot_1.1.6    lubridate_1.9.2  forcats_1.0.0    stringr_1.5.0   
 [5] dplyr_1.1.2      purrr_1.0.1      readr_2.1.4      tidyr_1.3.0     
 [9] tibble_3.2.1     tidyverse_2.0.0  lavaan_0.6-15    mediation_4.5.0 
[13] sandwich_3.0-2   mvtnorm_1.1-3    Matrix_1.5-4     MASS_7.3-58.3   
[17] GGally_2.1.2     factoextra_1.0.7 corrplot_0.92    ggfortify_0.4.16
[21] ggpubr_0.6.0     ggridges_0.5.4   ggplot2_3.4.2    readxl_1.4.2    
[25] workflowr_1.7.0 

loaded via a namespace (and not attached):
  [1] backports_1.4.1    Hmisc_5.1-0        igraph_1.5.0      
  [4] plyr_1.8.8         splines_4.2.2      digest_0.6.31     
  [7] htmltools_0.5.5    fansi_1.0.4        magrittr_2.0.3    
 [10] checkmate_2.2.0    lisrelToR_0.1.5    cluster_2.1.4     
 [13] tzdb_0.3.0         openxlsx_4.2.5.2   RcppParallel_5.1.7
 [16] vroom_1.6.1        timechange_0.2.0   lpSolve_5.6.18    
 [19] jpeg_0.1-10        sem_3.1-15         colorspace_2.1-0  
 [22] ggrepel_0.9.3      xfun_0.39          callr_3.7.3       
 [25] crayon_1.5.2       jsonlite_1.8.4     lme4_1.1-32       
 [28] zoo_1.8-12         glue_1.6.2         gtable_0.3.3      
 [31] mi_1.1             car_3.1-2          abind_1.4-5       
 [34] scales_1.2.1       rstatix_0.7.2      Rcpp_1.0.10       
 [37] xtable_1.8-4       htmlTable_2.4.1    foreign_0.8-84    
 [40] bit_4.0.5          Formula_1.2-5      stats4_4.2.2      
 [43] htmlwidgets_1.6.2  httr_1.4.5         RColorBrewer_1.1-3
 [46] pkgconfig_2.0.3    reshape_0.8.9      XML_3.99-0.14     
 [49] farver_2.1.1       nnet_7.3-18        sass_0.4.5        
 [52] kutils_1.72        utf8_1.2.3         reshape2_1.4.4    
 [55] tidyselect_1.2.0   labeling_0.4.2     rlang_1.1.0       
 [58] later_1.3.0        munsell_0.5.0      cellranger_1.1.0  
 [61] tools_4.2.2        cachem_1.0.7       cli_3.6.1         
 [64] generics_0.1.3     broom_1.0.4        fdrtool_1.2.17    
 [67] evaluate_0.20      fastmap_1.1.1      arm_1.13-1        
 [70] yaml_2.3.7         processx_3.8.1     knitr_1.42        
 [73] bit64_4.0.5        fs_1.6.1           zip_2.3.0         
 [76] glasso_1.11        pbapply_1.7-0      nlme_3.1-162      
 [79] whisker_0.4.1      compiler_4.2.2     rstudioapi_0.14   
 [82] png_0.1-8          ggsignif_0.6.4     bslib_0.4.2       
 [85] pbivnorm_0.6.0     stringi_1.7.12     highr_0.10        
 [88] ps_1.7.5           qgraph_1.9.5       rockchalk_1.8.157 
 [91] lattice_0.21-8     psych_2.3.6        nloptr_2.0.3      
 [94] ggsci_3.0.0        vctrs_0.6.2        pillar_1.9.0      
 [97] lifecycle_1.0.3    jquerylib_0.1.4    OpenMx_2.21.8     
[100] data.table_1.14.8  corpcor_1.6.10     httpuv_1.6.9      
[103] R6_2.5.1           promises_1.2.0.1   gridExtra_2.3     
[106] gtools_3.9.4       boot_1.3-28.1      rprojroot_2.0.3   
[109] withr_2.5.0        mnormt_2.1.1       parallel_4.2.2    
[112] hms_1.1.3          quadprog_1.5-8     grid_4.2.2        
[115] rpart_4.1.19       coda_0.19-4        minqa_1.2.5       
[118] rmarkdown_2.21     carData_3.0-5      git2r_0.32.0      
[121] getPass_0.2-2      base64enc_0.1-3   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
